<!DOCTYPE html>
<html lang="en"><head>
<script src="RStudio_paper_writing_course_presentation_part1_files/libs/clipboard/clipboard.min.js"></script>
<script src="RStudio_paper_writing_course_presentation_part1_files/libs/quarto-html/tabby.min.js"></script>
<script src="RStudio_paper_writing_course_presentation_part1_files/libs/quarto-html/popper.min.js"></script>
<script src="RStudio_paper_writing_course_presentation_part1_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="RStudio_paper_writing_course_presentation_part1_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="RStudio_paper_writing_course_presentation_part1_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="RStudio_paper_writing_course_presentation_part1_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="RStudio_paper_writing_course_presentation_part1_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.5.14">

  <meta name="author" content="Gáspár Jékely   Centre for Organismal Studies, Heidelberg University">
  <title>Rstudio in Data Science and Paper Writing</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; font-weight: bold; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/dist/theme/quarto.css">
  <link href="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/plugin/reveal-chalkboard/font-awesome/css/all.css" rel="stylesheet">
  <link href="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/plugin/reveal-chalkboard/style.css" rel="stylesheet">
  <link href="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    padding-bottom: 0.5rem;
    margin-bottom: 0;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Rstudio in Data Science and Paper Writing</h1>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Gáspár Jékely <br> Centre for Organismal Studies, Heidelberg University 
</div>
</div>
</div>

</section>
<section id="course-program" class="slide level2">
<h2>Course program</h2>
<p><br></p>
<ul>
<li><p><strong>Part 1</strong>: Why open science?<br>
</p></li>
<li><p><strong>Part 2</strong>: Project management, data import, tidy data<br>
</p></li>
<li><p><strong>Part 3</strong>: Plotting with ggplot<br>
</p></li>
<li><p><strong>Part 4</strong>: Assembly of figures<br>
</p></li>
<li><p><strong>Part 5</strong>: Manuscript writing, tables, references, data and figure embedding<br>
</p></li>
<li><p><strong>Part 6</strong>: GitHub, collaboration, sharing, version control</p></li>
</ul>
</section>
<section id="resources" class="slide level2">
<h2>Resources</h2>
<p><br><br></p>
<ul>
<li>The <a href="https://rstudio-education.github.io/tidyverse-cookbook/">Tidyverse Cookbook</a> (edited by Garrett Grolemund)<br>
</li>
</ul>
<p>https://rstudio-education.github.io/tidyverse-cookbook/</p>
</section>
<section id="part-1---why-do-we-need-open-science" class="slide level2">
<h2>Part 1 - Why do we need open science</h2>
<ul>
<li>Reproducibility crisis</li>
<li>Only a small fraction of research data is available</li>
<li>An even smaller fraction of code is available (physicists are notoriously bad in sharing)</li>
<li>Open access, if exists, is very expensive and maintains the profit of legacy publishers</li>
<li>Scholarly literature is antiquated, dysfunctional and rewards prestige/hype over quality and integrity</li>
<li>Scholarly workflows use non-professional, error-prone, closed-source software (MS, Adobe, Prism etc.) that makes sharing, integration, automation and collaboration difficult</li>
<li>The final product of years of research is often only a single pdf file (1990s tech) behind a paywall</li>
<li>Data, code and text are not searchable, reuseable, discoverable, shareable</li>
</ul>
</section>
<section id="most-source-data-collected-by-scientists-are-not-available" class="slide level2">
<h2>Most source data collected by scientists are not available</h2>
<p><br></p>
<p><img data-src="images/Data_publication_Pyramide.png" height="450"></p>
</section>
<section id="most-scientists-use-software-developed-for-accounting" class="slide level2">
<h2>Most scientists use software developed for accounting</h2>
<p><img data-src="images/Scientists_rename_genes.png"> <a href="https://www.theverge.com/2020/8/6/21355674/human-genes-rename-microsoft-excel-misreading-dates">theverge.com</a></p>
<ul>
<li>the symbol MARCH1 has now become MARCHF1, while SEPT1 has become SEPTIN1, and so on</li>
</ul>
<p><br></p>
</section>
<section id="gene-name-errors-are-widespread-in-the-scientific-literature" class="slide level2">
<h2>Gene name errors are widespread in the scientific literature</h2>
<p><br><br></p>
<p><img data-src="images/13059_2016_1044_Fig1_HTML.webp"></p>

<aside><div>
<p><a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-1044-7">Ziemann et al.&nbsp;(2016)</a></p>
</div></aside></section>
<section id="code-is-very-often-not-shared-or-not-shared-stably" class="slide level2">
<h2>Code is very often not shared or not shared stably</h2>
<ul>
<li><em>We assess the effectiveness of such a policy by (i) requesting data and code from authors and (ii) attempting replication of the published findings. We chose a random sample of 204 scientific papers published in the journal Science … We found that we were able to obtain artifacts from 44% of our sample and were able to reproduce the findings for 26%.</em></li>
</ul>
<div>
<ul>
<li class="fragment"><p>“When you approach a PI for the source codes and raw data, you better explain who you are, whom you work for, why you need the data and what you are going to do with it.”</p></li>
<li class="fragment"><p>“I have to say that this is a very unusual request without any explanation! Please ask your supervisor to send me an email with a detailed, and I mean detailed, explanation.”</p></li>
<li class="fragment"><p>“The data files remains our property and are not deposited for free access. Please, let me know the purpose you want to get the file and we will see how we can help you.”</p></li>
<li class="fragment"><p>“We do not typically share our internal data or code with people outside our collaboration.”</p></li>
</ul>
</div>
<p><img data-src="images/Stodden_PNAS.png" class="absolute" style="top: 480px; left: 300px; height: 200px; "></p>

<aside><div>
<p><a href="https://doi.org/10.1073/pnas.1708290115">Stodden et al.&nbsp;(2018)</a></p>
</div></aside></section>
<section id="current-state-of-scholarly-digital-infrastructure-and-knowhow" class="slide level2">
<h2>Current state of scholarly digital infrastructure and knowhow</h2>

<img src="https://media1.tenor.com/images/f6362876996697b6a6f554b2ac3d3013/tenor.gif?itemid=10488408" width="160" class="r-stretch"></section>
<section id="what-this-course-is-about-what-you-can-do-to-be-open" class="slide level2">
<h2>What this course is about: what you can do to be open</h2>
<ul>
<li>tools and approaches for transparent and open publishing</li>
<li>a paper is not only text and figures, but also data and code</li>
<li>all should be shared for reproducibility and openness</li>
<li>otherwise it is not really ‘published’</li>
<li>we will use Rstudio to learn a comprehensive paper-writing pipeline</li>
<li>code, tabulated data, figures, text, references, supplements all in one place</li>
<li>collaborative working and sharing via GitHub (or other public repository)</li>
<li>the figures are linked to their underlying data and code to generate them</li>
<li>efficient version control</li>
<li>faster, more transparent and reproducible workflow</li>
<li>no software license is needed (no MS, no Adobe, no Matlab)</li>
<li>once you master the approach I advocate, you don’t want to go back…</li>
</ul>
</section>
<section id="part-2---packages-project-management-data-import-tidy-data" class="slide level2">
<h2>Part 2 - Packages, project management, data import, tidy data</h2>
<p><br></p>
<ul>
<li>some rules on files, folders</li>
<li>R projects</li>
<li>installing and loading packages</li>
<li>project templates</li>
<li>importing various types of datasets</li>
<li>principles of tidy data, tidying of messy datasets</li>
</ul>
</section>
<section id="go-to-github-and-fork-my-repository" class="slide level2">
<h2>Go to GitHub and fork my repository</h2>
<p><br><br></p>
<ul>
<li>do you have a GitHub account?</li>
<li>navigate to: https://github.com/JekelyLab/RStudio_paper_writing_course</li>
<li>fork the repository to your GitHub (if you have an account already)</li>
<li>go to RStudio -&gt; New Project -&gt; Git -&gt; your repo’s address (or my address)</li>
<li>select local dir and download the project with all directories to your computer</li>
<li>let’s check the project</li>
<li>you can rename the .Rproject file to ‘my favourite file name’</li>
<li>open /analysis/scripts/Course_exercises.R</li>
</ul>
</section>
<section id="rule-nr.-1-relative-working-directories" class="slide level2">
<h2>Rule nr. 1 – relative working directories</h2>
<p><br></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href=""></a><span class="fu">getwd</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li><p>Never use absolute paths in your scripts, because they hinder sharing: no one else will have exactly the same directory configuration as you.</p></li>
<li><p>Do not use setwd() to set your working dir</p></li>
</ul>
</section>
<section id="use-r-projects-.rproj" class="slide level2">
<h2>Use R projects (.Rproj)</h2>
<p><br></p>
<ul>
<li><p>Keep all files associated with a project together — input data, R scripts, analytical results, figures. This is such a wise and common practice that RStudio has built-in support for this via <strong>Rprojects</strong>.</p></li>
<li><p>If you create a new Rproject, your working dir will in general be where you save the new project</p></li>
<li><p>Whenever you refer to a file with a relative path it will look for it in your working dir.</p></li>
<li><p>in case you used setwd() in a script (you should not) you can find your Rproject directory with here()</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href=""></a>here<span class="sc">::</span><span class="fu">here</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "/home/gj274/code/RStudio_paper_writing_course"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href=""></a><span class="co">#test, but you should not do this</span></span>
<span id="cb4-2"><a href=""></a><span class="fu">setwd</span>(<span class="st">"~"</span>)</span>
<span id="cb4-3"><a href=""></a><span class="fu">getwd</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "/home/gj274"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href=""></a><span class="co">#get back to Rproj dir</span></span>
<span id="cb6-2"><a href=""></a><span class="fu">setwd</span>(here<span class="sc">::</span><span class="fu">here</span>())</span>
<span id="cb6-3"><a href=""></a><span class="fu">getwd</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "/home/gj274/code/RStudio_paper_writing_course"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href=""></a><span class="co"># all good again</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="use-r-projects-.rproj-1" class="slide level2">
<h2>Use R projects (.Rproj)</h2>
<p><br></p>
<ul>
<li>list files in your working dir</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href=""></a><span class="fu">list.files</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "custom.scss"                                              
 [2] "images"                                                   
 [3] "references.bib"                                           
 [4] "RStudio_paper_writing_course_presentation_day1_files"     
 [5] "RStudio_paper_writing_course_presentation_day2_files"     
 [6] "RStudio_paper_writing_course_presentation_files"          
 [7] "RStudio_paper_writing_course_presentation_part1_files"    
 [8] "RStudio_paper_writing_course_presentation_part1.html"     
 [9] "RStudio_paper_writing_course_presentation_part1.qmd"      
[10] "RStudio_paper_writing_course_presentation_part1.rmarkdown"
[11] "RStudio_paper_writing_course_presentation_part2_files"    
[12] "RStudio_paper_writing_course_presentation_part2.html"     
[13] "RStudio_paper_writing_course_presentation_part2.qmd"      
[14] "RStudio_paper_writing_course_presentation.html"           
[15] "sessionInfo.txt"                                          </code></pre>
</div>
</div>
</section>
<section id="project-management" class="slide level2">
<h2>Project management</h2>
<p><br><br><br></p>
<ul>
<li><p>Use folders relative to your main .Rproject file (e.g.&nbsp;My_next_report.Rproject)</p></li>
<li><p>Use a consistent directory structure to store code, data, text, figures, supplements, etc.</p></li>
<li><p>Can be ensured if you always use the same project template</p></li>
<li><p>Check the folder structure in the downloaded project template</p></li>
</ul>
</section>
<section id="installing-packages" class="slide level2">
<h2>Installing packages</h2>
<ul>
<li>install the <strong>tidyverse</strong> package</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href=""></a><span class="fu">install.packages</span>(<span class="st">"tidyverse"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>then load the package</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href=""></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb12-2"><a href=""></a><span class="fu">library</span>(png)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="you-can-source-packages-and-functions-from-one-file" class="slide level2">
<h2>You can source packages and functions from one file</h2>
<p><br><br></p>
<ul>
<li><p>open the R script analysis/scripts/Course_exercises.R with the example code</p></li>
<li><p>source several packages and functions, listed in one file consistently across scripts</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href=""></a><span class="fu">source</span>(<span class="st">"../analysis/scripts/packages_and_functions.R"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="general-good-practice-avoid-spaces-in-file-names" class="slide level2">
<h2>General good practice – avoid spaces in file names</h2>
<div>
<p><br></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href=""></a><span class="fu">list.files</span>(<span class="st">"../analysis/data"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "24_01_22qpcr_1 - ash pal.xls"                  
 [2] "24_01_22qpcr_1 - ash pal.xlsx"                 
 [3] "240323 CIN Exp278 reporter assay - Anchel.xlsx"
 [4] "a-Syn-Data.csv"                                
 [5] "data - José - March 2024.xlsx"                 
 [6] "data_Jose_March2024.csv"                       
 [7] "data_Jose_March2024.csv.zip"                   
 [8] "head_celltypes_syn_matrix.csv"                 
 [9] "Key_resource_table.csv"                        
[10] "Key_resource_table.xls"                        
[11] "Key_resource_table.xlsx"                       
[12] "Tom_filopodia_analyses.csv"                    
[13] "Tom_filopodia_analyses.csv~"                   </code></pre>
</div>
</div>
<p><br></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href=""></a>readxl<span class="sc">::</span><span class="fu">read_excel</span>(<span class="st">"../analysis/data/data - José - March 2024.xlsx"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 692 × 3
   Treatment genotype       length
   &lt;chr&gt;     &lt;chr&gt;           &lt;dbl&gt;
 1 Control   CLE25_OE_1.1#2   2.19
 2 Control   CLE25_OE_1.1#2   2.62
 3 Control   CLE25_OE_1.1#2   2.67
 4 Control   CLE25_OE_1.1#2   2.28
 5 Control   CLE25_OE_1.1#2   2.21
 6 Control   CLE25_OE_1.1#2   2.83
 7 Control   CLE25_OE_1.1#2   2.13
 8 Control   CLE25_OE_1.1#2   2.76
 9 Control   CLE25_OE_1.1#2   2.88
10 Control   CLE25_OE_1.1#2   2.95
# ℹ 682 more rows</code></pre>
</div>
</div>
</div>
</section>
<section id="save-and-share-your-computer-environment-and-packages" class="slide level2">
<h2>Save and share your computer environment and packages</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href=""></a><span class="co">#save session info and Rstudio version info for reproducibility</span></span>
<span id="cb18-2"><a href=""></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 24.04.1 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.12.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.12.0

locale:
 [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
 [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
 [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       

time zone: Europe/Berlin
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] rio_1.2.3          tinytable_0.5.0    janitor_2.2.0      magick_2.8.5      
 [5] rstudioapi_0.17.1  ggpubr_0.6.0       RColorBrewer_1.1-3 patchwork_1.3.0   
 [9] png_0.1-8          cowplot_1.1.3      lubridate_1.9.3    forcats_1.0.0     
[13] stringr_1.5.1      dplyr_1.1.4        purrr_1.0.2        readr_2.1.5       
[17] tidyr_1.3.1        tibble_3.2.1       ggplot2_3.5.1      tidyverse_2.0.0   

loaded via a namespace (and not attached):
 [1] utf8_1.2.4        generics_0.1.3    rstatix_0.7.2     stringi_1.8.4    
 [5] hms_1.1.3         digest_0.6.37     magrittr_2.0.3    evaluate_1.0.1   
 [9] grid_4.4.1        timechange_0.3.0  fastmap_1.2.0     cellranger_1.1.0 
[13] rprojroot_2.0.4   jsonlite_1.8.9    backports_1.5.0   Formula_1.2-5    
[17] fansi_1.0.6       scales_1.3.0      abind_1.4-8       cli_3.6.3        
[21] rlang_1.1.4       munsell_0.5.1     withr_3.0.2       yaml_2.3.10      
[25] tools_4.4.1       tzdb_0.4.0        ggsignif_0.6.4    colorspace_2.1-1 
[29] here_1.0.1        broom_1.0.7       vctrs_0.6.5       R6_2.5.1         
[33] lifecycle_1.0.4   snakecase_0.11.1  car_3.1-3         pkgconfig_2.0.3  
[37] pillar_1.9.0      gtable_0.3.6      Rcpp_1.0.13       glue_1.8.0       
[41] xfun_0.48         tidyselect_1.2.1  knitr_1.48        farver_2.1.2     
[45] htmltools_0.5.8.1 carData_3.0-5     rmarkdown_2.28    compiler_4.4.1   
[49] readxl_1.4.3     </code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href=""></a><span class="fu">writeLines</span>(<span class="fu">capture.output</span>(<span class="fu">sessionInfo</span>()), <span class="st">"sessionInfo.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="file-import-and-saving-rio-a-swiss-army-knife-for-data-io" class="slide level2">
<h2>File import and saving – rio: A Swiss-Army Knife for Data I/O</h2>
<ul>
<li>import xlsx, csv, tsv and many other formats with the same command</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href=""></a>data_Jose <span class="ot">&lt;-</span> rio<span class="sc">::</span><span class="fu">import</span>(<span class="st">"../analysis/data/data - José - March 2024.xlsx"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>export as csv</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href=""></a>rio<span class="sc">::</span><span class="fu">export</span>(data_Jose, <span class="st">"../analysis/data/data_Jose_March2024.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>export as compressed csv</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href=""></a><span class="fu">export</span>(data_Jose, <span class="st">"../analysis/data/data_Jose_March2024.csv.zip"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>convert file formats</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href=""></a>rio<span class="sc">::</span><span class="fu">convert</span>(<span class="st">"../analysis/data/data - José - March 2024.xlsx"</span>,</span>
<span id="cb24-2"><a href=""></a><span class="st">"../analysis/data/data_Jose_March2024.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>

<aside><div>
<p><a href="http://gesistsa.github.io/rio/">Documentation of the rio package</a></p>
</div></aside></section>
<section id="read-and-preview-data-1" class="slide level2">
<h2>Read and preview data 1</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href=""></a>data_Jose <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_excel</span>(<span class="st">"../analysis/data/data - José - March 2024.xlsx"</span>)</span>
<span id="cb25-2"><a href=""></a><span class="fu">head</span>(data_Jose)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 3
  Treatment genotype       length
  &lt;chr&gt;     &lt;chr&gt;           &lt;dbl&gt;
1 Control   CLE25_OE_1.1#2   2.19
2 Control   CLE25_OE_1.1#2   2.62
3 Control   CLE25_OE_1.1#2   2.67
4 Control   CLE25_OE_1.1#2   2.28
5 Control   CLE25_OE_1.1#2   2.21
6 Control   CLE25_OE_1.1#2   2.83</code></pre>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href=""></a><span class="fu">glimpse</span>(data_Jose)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 692
Columns: 3
$ Treatment &lt;chr&gt; "Control", "Control", "Control", "Control", "Control", "Cont…
$ genotype  &lt;chr&gt; "CLE25_OE_1.1#2", "CLE25_OE_1.1#2", "CLE25_OE_1.1#2", "CLE25…
$ length    &lt;dbl&gt; 2.190, 2.622, 2.673, 2.276, 2.209, 2.829, 2.130, 2.761, 2.88…</code></pre>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href=""></a><span class="fu">str</span>(data_Jose)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>tibble [692 × 3] (S3: tbl_df/tbl/data.frame)
 $ Treatment: chr [1:692] "Control" "Control" "Control" "Control" ...
 $ genotype : chr [1:692] "CLE25_OE_1.1#2" "CLE25_OE_1.1#2" "CLE25_OE_1.1#2" "CLE25_OE_1.1#2" ...
 $ length   : num [1:692] 2.19 2.62 2.67 2.28 2.21 ...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href=""></a><span class="fu">summary</span>(data_Jose)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  Treatment           genotype             length     
 Length:692         Length:692         Min.   :0.948  
 Class :character   Class :character   1st Qu.:2.071  
 Mode  :character   Mode  :character   Median :2.566  
                                       Mean   :2.746  
                                       3rd Qu.:3.270  
                                       Max.   :7.981  </code></pre>
</div>
</div>
</section>
<section id="your-workspace-in-r" class="slide level2">
<h2>Your workspace in R</h2>
<ul>
<li>Recreate, rather than save workspace, save your coda and data, not workspace <br></li>
</ul>

<img src="images/rstudio-workspace.png" width="60%" class="r-stretch"></section>
<section id="read-and-preview-data-2" class="slide level2">
<h2>Read and preview data 2</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href=""></a>data_Ashwini <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_excel</span>(<span class="st">"../analysis/data/24_01_22qpcr_1 - ash pal.xlsx"</span>)</span>
<span id="cb33-2"><a href=""></a><span class="fu">head</span>(data_Ashwini)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 23
  Gene  Days  `Raw Data` `Ct Value`         `Dt Ct` `2^(-dct)` `MEAN 2^(-dCt)`
  &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;      &lt;chr&gt;                &lt;dbl&gt;      &lt;dbl&gt;           &lt;dbl&gt;
1 Nanog d0    J1         22.491                6.96    0.00805          0.0809
2 &lt;NA&gt;  d0    J2         19.876999999999999    4.40    0.0475          NA     
3 &lt;NA&gt;  d0    J3         19.164000000000001    2.42    0.187           NA     
4 &lt;NA&gt;  d7    J4         21.177                4.67    0.0393           0.0319
5 &lt;NA&gt;  d7    J5         21.812000000000001    4.93    0.0327          NA     
6 &lt;NA&gt;  d7    J6         22.303000000000001    5.40    0.0237          NA     
# ℹ 16 more variables: `SD 2^(-dCt)` &lt;dbl&gt;, ...9 &lt;lgl&gt;, ...10 &lt;lgl&gt;,
#   ...11 &lt;chr&gt;, ...12 &lt;chr&gt;, ...13 &lt;chr&gt;, ...14 &lt;chr&gt;, ...15 &lt;chr&gt;,
#   ...16 &lt;chr&gt;, ...17 &lt;chr&gt;, ...18 &lt;chr&gt;, ...19 &lt;chr&gt;, ...20 &lt;chr&gt;,
#   ...21 &lt;chr&gt;, ...22 &lt;chr&gt;, ...23 &lt;chr&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href=""></a><span class="fu">glimpse</span>(data_Ashwini)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 105
Columns: 23
$ Gene            &lt;chr&gt; "Nanog", NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ Days            &lt;chr&gt; "d0", "d0", "d0", "d7", "d7", "d7", "d11", "d11", "d11…
$ `Raw Data`      &lt;chr&gt; "J1", "J2", "J3", "J4", "J5", "J6", "J7", "J8", "J9", …
$ `Ct Value`      &lt;chr&gt; "22.491", "19.876999999999999", "19.164000000000001", …
$ `Dt Ct`         &lt;dbl&gt; 6.956, 4.396, 2.418, 4.668, 4.934, 5.397, 7.436, 7.413…
$ `2^(-dct)`      &lt;dbl&gt; 0.008054440, 0.047497652, 0.187115373, 0.039336162, 0.…
$ `MEAN 2^(-dCt)` &lt;dbl&gt; 0.08088916, NA, NA, 0.03192712, NA, NA, 0.14529882, NA…
$ `SD 2^(-dCt)`   &lt;dbl&gt; 0.094084796, NA, NA, 0.007831512, NA, NA, 0.241582234,…
$ ...9            &lt;lgl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA…
$ ...10           &lt;lgl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA…
$ ...11           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, "d0", "d7", "d11",…
$ ...12           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, "Nanog", NA, "8.0889155034…
$ ...13           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, "SD", "9.4084795965414…
$ ...14           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, "oct4", NA, "0.31103134803…
$ ...15           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, "SD", "0.1837141821452…
$ ...16           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, "sox2", NA, "1.21068833126…
$ ...17           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, "SD", "5.0018821338136…
$ ...18           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, "Nestin", NA, "1.005919450…
$ ...19           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, "SD", "3.7803772492582…
$ ...20           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, "pax6", NA, "2.02563640616…
$ ...21           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, "SD", "9.0704917358939…
$ ...22           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, "Foxg1", NA, "2.8294514961…
$ ...23           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, "SD", "1.7496262363729…</code></pre>
</div>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href=""></a><span class="fu">str</span>(data_Ashwini)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>tibble [105 × 23] (S3: tbl_df/tbl/data.frame)
 $ Gene         : chr [1:105] "Nanog" NA NA NA ...
 $ Days         : chr [1:105] "d0" "d0" "d0" "d7" ...
 $ Raw Data     : chr [1:105] "J1" "J2" "J3" "J4" ...
 $ Ct Value     : chr [1:105] "22.491" "19.876999999999999" "19.164000000000001" "21.177" ...
 $ Dt Ct        : num [1:105] 6.96 4.4 2.42 4.67 4.93 ...
 $ 2^(-dct)     : num [1:105] 0.00805 0.0475 0.18712 0.03934 0.03271 ...
 $ MEAN 2^(-dCt): num [1:105] 0.0809 NA NA 0.0319 NA ...
 $ SD 2^(-dCt)  : num [1:105] 0.09408 NA NA 0.00783 NA ...
 $ ...9         : logi [1:105] NA NA NA NA NA NA ...
 $ ...10        : logi [1:105] NA NA NA NA NA NA ...
 $ ...11        : chr [1:105] NA NA NA NA ...
 $ ...12        : chr [1:105] NA NA NA NA ...
 $ ...13        : chr [1:105] NA NA NA NA ...
 $ ...14        : chr [1:105] NA NA NA NA ...
 $ ...15        : chr [1:105] NA NA NA NA ...
 $ ...16        : chr [1:105] NA NA NA NA ...
 $ ...17        : chr [1:105] NA NA NA NA ...
 $ ...18        : chr [1:105] NA NA NA NA ...
 $ ...19        : chr [1:105] NA NA NA NA ...
 $ ...20        : chr [1:105] NA NA NA NA ...
 $ ...21        : chr [1:105] NA NA NA NA ...
 $ ...22        : chr [1:105] NA NA NA NA ...
 $ ...23        : chr [1:105] NA NA NA NA ...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href=""></a><span class="fu">summary</span>(data_Ashwini)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     Gene               Days             Raw Data           Ct Value        
 Length:105         Length:105         Length:105         Length:105        
 Class :character   Class :character   Class :character   Class :character  
 Mode  :character   Mode  :character   Mode  :character   Mode  :character  
                                                                            
                                                                            
                                                                            
                                                                            
     Dt Ct           2^(-dct)         MEAN 2^(-dCt)      SD 2^(-dCt)     
 Min.   : 0.000   Min.   :0.0001406   Min.   :0.00020   Min.   :0.00000  
 1st Qu.: 2.853   1st Qu.:0.0009202   1st Qu.:0.00101   1st Qu.:0.00015  
 Median : 6.906   Median :0.0083435   Median :0.03660   Median :0.00123  
 Mean   : 6.362   Mean   :0.1927351   Mean   :0.19091   Mean   :0.05215  
 3rd Qu.:10.086   3rd Qu.:0.1384743   3rd Qu.:0.13341   3rd Qu.:0.04960  
 Max.   :12.796   Max.   :1.0000000   Max.   :1.00000   Max.   :0.49862  
 NA's   :1        NA's   :1           NA's   :70        NA's   :70       
   ...9          ...10            ...11              ...12          
 Mode:logical   Mode:logical   Length:105         Length:105        
 NA's:105       NA's:105       Class :character   Class :character  
                               Mode  :character   Mode  :character  
                                                                    
                                                                    
                                                                    
                                                                    
    ...13              ...14              ...15              ...16          
 Length:105         Length:105         Length:105         Length:105        
 Class :character   Class :character   Class :character   Class :character  
 Mode  :character   Mode  :character   Mode  :character   Mode  :character  
                                                                            
                                                                            
                                                                            
                                                                            
    ...17              ...18              ...19              ...20          
 Length:105         Length:105         Length:105         Length:105        
 Class :character   Class :character   Class :character   Class :character  
 Mode  :character   Mode  :character   Mode  :character   Mode  :character  
                                                                            
                                                                            
                                                                            
                                                                            
    ...21              ...22              ...23          
 Length:105         Length:105         Length:105        
 Class :character   Class :character   Class :character  
 Mode  :character   Mode  :character   Mode  :character  
                                                         
                                                         
                                                         
                                                         </code></pre>
</div>
</div>
</section>
<section id="tidy-data---definitions" class="slide level2">
<h2>Tidy data - definitions</h2>
<p>from https://r4ds.hadley.nz/data-visualize</p>
<ul>
<li>A <strong>variable</strong> is a quantity, quality, or property that you can measure</li>
<li>A <strong>value</strong> is the state of a variable when you measure it. The value of a variable may change from measurement to measurement.</li>
<li>An <strong>observation</strong> is a set of measurements made under similar conditions (you usually make all of the measurements in an observation at the same time and on the same object). An observation will contain several values, each associated with a different variable. We’ll sometimes refer to an observation as a data point.</li>
</ul>
</section>
<section id="tidy-data---definitions-1" class="slide level2">
<h2>Tidy data - definitions</h2>
<ul>
<li><p>Tabular data is a set of values, each associated with a variable and an observation.</p></li>
<li><p>Tabular data is tidy if each value is placed in its own “cell”, each variable in its own column, and each observation in its own row.</p></li>
<li><p>“Tidy datasets are all alike, but every messy dataset is messy in its own way.” –– Hadley Wickham</p></li>
</ul>
<p><strong>There are three interrelated rules which make a dataset tidy:</strong></p>
<ul>
<li>Each variable must have its own column.</li>
<li>Each observation must have its own row.</li>
<li>Each value must have its own cell.</li>
</ul>

<img src="images/tidy-1.png" width="60%" class="r-stretch"></section>
<section id="tidy-data---an-example-tidy-dataset" class="slide level2">
<h2>Tidy data - an example tidy dataset</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href=""></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(iris), <span class="at">format =</span> <span class="st">'html'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<table data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: right;" data-quarto-table-cell-role="th">Sepal.Length</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Sepal.Width</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Petal.Length</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Petal.Width</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Species</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">5.1</td>
<td style="text-align: right;">3.5</td>
<td style="text-align: right;">1.4</td>
<td style="text-align: right;">0.2</td>
<td style="text-align: left;">setosa</td>
</tr>
<tr class="even">
<td style="text-align: right;">4.9</td>
<td style="text-align: right;">3.0</td>
<td style="text-align: right;">1.4</td>
<td style="text-align: right;">0.2</td>
<td style="text-align: left;">setosa</td>
</tr>
<tr class="odd">
<td style="text-align: right;">4.7</td>
<td style="text-align: right;">3.2</td>
<td style="text-align: right;">1.3</td>
<td style="text-align: right;">0.2</td>
<td style="text-align: left;">setosa</td>
</tr>
<tr class="even">
<td style="text-align: right;">4.6</td>
<td style="text-align: right;">3.1</td>
<td style="text-align: right;">1.5</td>
<td style="text-align: right;">0.2</td>
<td style="text-align: left;">setosa</td>
</tr>
<tr class="odd">
<td style="text-align: right;">5.0</td>
<td style="text-align: right;">3.6</td>
<td style="text-align: right;">1.4</td>
<td style="text-align: right;">0.2</td>
<td style="text-align: left;">setosa</td>
</tr>
<tr class="even">
<td style="text-align: right;">5.4</td>
<td style="text-align: right;">3.9</td>
<td style="text-align: right;">1.7</td>
<td style="text-align: right;">0.4</td>
<td style="text-align: left;">setosa</td>
</tr>
</tbody>
</table>


</div>
</div>
</div>
</section>
<section id="tibbles-tabular-data-in-tidy-format" class="slide level2">
<h2>Tibbles = tabular data in tidy format</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href=""></a><span class="fu">vignette</span>(<span class="st">"tibble"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Tibbles are a modern take on data frames.</li>
<li>They keep the features that have stood the test of time, and drop the features that used to be convenient but are now frustrating.</li>
<li>Tibbles encapsulate best practices for data frames.</li>
<li>tibble() It never uses row.names().</li>
</ul>
</section>
<section id="tidy-data-why-should-i-care" class="slide level2">
<h2>Tidy data – <strong>Why should I care?</strong></h2>
<p><br></p>
<ul>
<li>after reading your data, you should always try to convert them into a tibble</li>
<li>downstream analyses (plotting, mutating, sharing etc.) will be a lot easier</li>
<li>data coming from other software, collaborators etc. are often ‘messy’</li>
<li>it is worth investing the time in tidying the data first</li>
<li>sometimes it is tricky, as we will see…</li>
</ul>
</section>
<section id="lets-try-to-fix-a-messy-dataset" class="slide level2">
<h2>Let’s try to fix a messy dataset…</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href=""></a>Gene <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"Nanog"</span>, <span class="dv">15</span>), <span class="fu">rep</span>(<span class="st">"oct4"</span>, <span class="dv">15</span>), <span class="fu">rep</span>(<span class="st">"sox2"</span>, <span class="dv">15</span>),</span>
<span id="cb43-2"><a href=""></a>  <span class="fu">rep</span>(<span class="st">"Nestin"</span>, <span class="dv">15</span>), <span class="fu">rep</span>(<span class="st">"pax6"</span>, <span class="dv">15</span>), <span class="fu">rep</span>(<span class="st">"Foxg1"</span>, <span class="dv">15</span>),</span>
<span id="cb43-3"><a href=""></a>  <span class="fu">rep</span>(<span class="st">"GAPDH"</span>, <span class="dv">15</span>))</span>
<span id="cb43-4"><a href=""></a>Gene</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  [1] "Nanog"  "Nanog"  "Nanog"  "Nanog"  "Nanog"  "Nanog"  "Nanog"  "Nanog" 
  [9] "Nanog"  "Nanog"  "Nanog"  "Nanog"  "Nanog"  "Nanog"  "Nanog"  "oct4"  
 [17] "oct4"   "oct4"   "oct4"   "oct4"   "oct4"   "oct4"   "oct4"   "oct4"  
 [25] "oct4"   "oct4"   "oct4"   "oct4"   "oct4"   "oct4"   "sox2"   "sox2"  
 [33] "sox2"   "sox2"   "sox2"   "sox2"   "sox2"   "sox2"   "sox2"   "sox2"  
 [41] "sox2"   "sox2"   "sox2"   "sox2"   "sox2"   "Nestin" "Nestin" "Nestin"
 [49] "Nestin" "Nestin" "Nestin" "Nestin" "Nestin" "Nestin" "Nestin" "Nestin"
 [57] "Nestin" "Nestin" "Nestin" "Nestin" "pax6"   "pax6"   "pax6"   "pax6"  
 [65] "pax6"   "pax6"   "pax6"   "pax6"   "pax6"   "pax6"   "pax6"   "pax6"  
 [73] "pax6"   "pax6"   "pax6"   "Foxg1"  "Foxg1"  "Foxg1"  "Foxg1"  "Foxg1" 
 [81] "Foxg1"  "Foxg1"  "Foxg1"  "Foxg1"  "Foxg1"  "Foxg1"  "Foxg1"  "Foxg1" 
 [89] "Foxg1"  "Foxg1"  "GAPDH"  "GAPDH"  "GAPDH"  "GAPDH"  "GAPDH"  "GAPDH" 
 [97] "GAPDH"  "GAPDH"  "GAPDH"  "GAPDH"  "GAPDH"  "GAPDH"  "GAPDH"  "GAPDH" 
[105] "GAPDH" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href=""></a>data_Ashwini<span class="sc">$</span>Gene <span class="ot">&lt;-</span> Gene</span>
<span id="cb45-2"><a href=""></a><span class="fu">head</span>(data_Ashwini)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 23
  Gene  Days  `Raw Data` `Ct Value`         `Dt Ct` `2^(-dct)` `MEAN 2^(-dCt)`
  &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;      &lt;chr&gt;                &lt;dbl&gt;      &lt;dbl&gt;           &lt;dbl&gt;
1 Nanog d0    J1         22.491                6.96    0.00805          0.0809
2 Nanog d0    J2         19.876999999999999    4.40    0.0475          NA     
3 Nanog d0    J3         19.164000000000001    2.42    0.187           NA     
4 Nanog d7    J4         21.177                4.67    0.0393           0.0319
5 Nanog d7    J5         21.812000000000001    4.93    0.0327          NA     
6 Nanog d7    J6         22.303000000000001    5.40    0.0237          NA     
# ℹ 16 more variables: `SD 2^(-dCt)` &lt;dbl&gt;, ...9 &lt;lgl&gt;, ...10 &lt;lgl&gt;,
#   ...11 &lt;chr&gt;, ...12 &lt;chr&gt;, ...13 &lt;chr&gt;, ...14 &lt;chr&gt;, ...15 &lt;chr&gt;,
#   ...16 &lt;chr&gt;, ...17 &lt;chr&gt;, ...18 &lt;chr&gt;, ...19 &lt;chr&gt;, ...20 &lt;chr&gt;,
#   ...21 &lt;chr&gt;, ...22 &lt;chr&gt;, ...23 &lt;chr&gt;</code></pre>
</div>
</div>
</section>
<section id="select-only-relevant-columns-and-clean-up-names" class="slide level2">
<h2>Select only relevant columns and clean up names…</h2>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 105 × 6
   gene  days  raw_data ct_value           dt_ct  x2_dct
   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;    &lt;chr&gt;              &lt;dbl&gt;   &lt;dbl&gt;
 1 Nanog d0    J1       22.491              6.96 0.00805
 2 Nanog d0    J2       19.876999999999999  4.40 0.0475 
 3 Nanog d0    J3       19.164000000000001  2.42 0.187  
 4 Nanog d7    J4       21.177              4.67 0.0393 
 5 Nanog d7    J5       21.812000000000001  4.93 0.0327 
 6 Nanog d7    J6       22.303000000000001  5.40 0.0237 
 7 Nanog d11   J7       22.93               7.44 0.00577
 8 Nanog d11   J8       22.914999999999999  7.41 0.00587
 9 Nanog d11   J9       17.831              1.24 0.424  
10 Nanog d15   J10      15.848000000000001  1.25 0.420  
# ℹ 95 more rows</code></pre>
</div>
</div>
</section>
<section id="add-mean-and-sd-columns-with-group_by-and-mutate" class="slide level2">
<h2>Add mean and SD columns with group_by() and mutate()</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href=""></a>data_Ashwini_sel_M_SD <span class="ot">&lt;-</span> data_Ashwini_sel <span class="sc">%&gt;%</span></span>
<span id="cb48-2"><a href=""></a>  <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span></span>
<span id="cb48-3"><a href=""></a>  <span class="fu">mutate</span>(<span class="at">mean2dct =</span> <span class="fu">mean</span>(x2_dct)) <span class="sc">%&gt;%</span></span>
<span id="cb48-4"><a href=""></a>  <span class="fu">mutate</span>(<span class="at">sd2dct =</span> <span class="fu">sd</span>(x2_dct))</span>
<span id="cb48-5"><a href=""></a>data_Ashwini_sel_M_SD</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 105 × 8
# Groups:   gene [7]
   gene  days  raw_data ct_value           dt_ct  x2_dct mean2dct sd2dct
   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;    &lt;chr&gt;              &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;  &lt;dbl&gt;
 1 Nanog d0    J1       22.491              6.96 0.00805   0.0909  0.141
 2 Nanog d0    J2       19.876999999999999  4.40 0.0475    0.0909  0.141
 3 Nanog d0    J3       19.164000000000001  2.42 0.187     0.0909  0.141
 4 Nanog d7    J4       21.177              4.67 0.0393    0.0909  0.141
 5 Nanog d7    J5       21.812000000000001  4.93 0.0327    0.0909  0.141
 6 Nanog d7    J6       22.303000000000001  5.40 0.0237    0.0909  0.141
 7 Nanog d11   J7       22.93               7.44 0.00577   0.0909  0.141
 8 Nanog d11   J8       22.914999999999999  7.41 0.00587   0.0909  0.141
 9 Nanog d11   J9       17.831              1.24 0.424     0.0909  0.141
10 Nanog d15   J10      15.848000000000001  1.25 0.420     0.0909  0.141
# ℹ 95 more rows</code></pre>
</div>
</div>
</section>
<section id="change-data-type" class="slide level2">
<h2>Change data type…</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href=""></a>data_Ashwini_sel_M_SD <span class="ot">&lt;-</span> data_Ashwini_sel_M_SD <span class="sc">%&gt;%</span></span>
<span id="cb50-2"><a href=""></a>  <span class="fu">mutate</span>(<span class="at">ct_value =</span> <span class="fu">as.double</span>(ct_value))</span>
<span id="cb50-3"><a href=""></a>data_Ashwini_sel_M_SD</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 105 × 8
# Groups:   gene [7]
   gene  days  raw_data ct_value dt_ct  x2_dct mean2dct sd2dct
   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;  &lt;dbl&gt;
 1 Nanog d0    J1           22.5  6.96 0.00805   0.0909  0.141
 2 Nanog d0    J2           19.9  4.40 0.0475    0.0909  0.141
 3 Nanog d0    J3           19.2  2.42 0.187     0.0909  0.141
 4 Nanog d7    J4           21.2  4.67 0.0393    0.0909  0.141
 5 Nanog d7    J5           21.8  4.93 0.0327    0.0909  0.141
 6 Nanog d7    J6           22.3  5.40 0.0237    0.0909  0.141
 7 Nanog d11   J7           22.9  7.44 0.00577   0.0909  0.141
 8 Nanog d11   J8           22.9  7.41 0.00587   0.0909  0.141
 9 Nanog d11   J9           17.8  1.24 0.424     0.0909  0.141
10 Nanog d15   J10          15.8  1.25 0.420     0.0909  0.141
# ℹ 95 more rows</code></pre>
</div>
</div>
</section>
<section id="pivoting" class="slide level2">
<h2>Pivoting</h2>
<ul>
<li>Tidying messy data is hard and requires parctice</li>
<li>Read more here: https://r4ds.had.co.nz/tibbles.html</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href=""></a>data_Syn <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../analysis/data/a-Syn-Data.csv"</span>)</span>
<span id="cb52-2"><a href=""></a>data_Syn</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 368 × 13
    Time aSyn...2 aSyn...3 aSyn...4 `aSyn-DMSO...5` `aSyn-DMSO...6`
   &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;           &lt;dbl&gt;           &lt;dbl&gt;
 1 0         655.     639.     687.            265.            381.
 2 0.264    1151.    1006.     863.            793.            860.
 3 0.527    1173     1104     1082             910.           1222.
 4 0.791    2542.    2191.    2451.           1062            1273 
 5 1.05     3922     3573     3790            1284            1385 
 6 1.32     4716.    4638.    4670.           2456.           2869.
 7 1.58     5322.    5424.    5323.           2955.           3986.
 8 1.85     5813     6041     5794            3661            4463 
 9 2.11     5983.    6325.    6227.           4345.           5133.
10 2.37     6147.    6584.    6351.           4369.           5363.
# ℹ 358 more rows
# ℹ 7 more variables: `aSyn-DMSO...7` &lt;dbl&gt;, aSyn_D2_13uM...8 &lt;dbl&gt;,
#   aSyn_D2_13uM...9 &lt;dbl&gt;, aSyn_D2_13uM...10 &lt;dbl&gt;, aSyn_D3_0.67mM...11 &lt;dbl&gt;,
#   aSyn_D3_0.67mM...12 &lt;dbl&gt;, aSyn_D3_0.67mM...13 &lt;dbl&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href=""></a><span class="co"># rename</span></span>
<span id="cb54-2"><a href=""></a>data_Syn_clean <span class="ot">&lt;-</span> data_Syn  <span class="sc">%&gt;%</span></span>
<span id="cb54-3"><a href=""></a>  <span class="fu">rename_with</span>(<span class="sc">~</span> <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">"-"</span>, .x, <span class="at">fixed =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-4"><a href=""></a>  <span class="fu">rename_with</span>(<span class="sc">~</span> <span class="fu">gsub</span>(<span class="st">"..."</span>, <span class="st">"_"</span>, .x, <span class="at">fixed =</span> <span class="cn">TRUE</span>))</span>
<span id="cb54-5"><a href=""></a></span>
<span id="cb54-6"><a href=""></a>tb_syn <span class="ot">&lt;-</span> data_Syn_clean <span class="sc">|&gt;</span></span>
<span id="cb54-7"><a href=""></a>  <span class="fu">pivot_longer</span>(<span class="fu">matches</span>(<span class="st">"aSyn"</span>), </span>
<span id="cb54-8"><a href=""></a>               <span class="at">names_to =</span> <span class="fu">c</span>(<span class="st">"condition"</span>, <span class="st">"sample"</span>), </span>
<span id="cb54-9"><a href=""></a>               <span class="at">names_sep =</span> <span class="st">"_"</span>,</span>
<span id="cb54-10"><a href=""></a>               <span class="at">values_to =</span> <span class="st">"fluorescence"</span>)</span>
<span id="cb54-11"><a href=""></a> tb_syn</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 4,416 × 4
    Time condition      sample fluorescence
   &lt;dbl&gt; &lt;chr&gt;          &lt;chr&gt;         &lt;dbl&gt;
 1     0 aSyn           2              655.
 2     0 aSyn           3              639.
 3     0 aSyn           4              687.
 4     0 aSyn-DMSO      5              265.
 5     0 aSyn-DMSO      6              381.
 6     0 aSyn-DMSO      7              424.
 7     0 aSyn-D2-13uM   8              436 
 8     0 aSyn-D2-13uM   9              461 
 9     0 aSyn-D2-13uM   10             672 
10     0 aSyn-D3-0.67mM 11             572.
# ℹ 4,406 more rows</code></pre>
</div>
</div>
</section>
<section id="part-3---data-plotting-with-ggplot" class="slide level2">
<h2>Part 3 - data plotting with ggplot</h2>
<p><br></p>
<ul>
<li>once your data have been tidied up, plotting is a lot easier</li>
<li>use ggplot2 and a tibble as input</li>
<li>ggplot2 is part of tidyverse and loads when you load tidyverse</li>
<li>very versatile and extendable data visualisation package</li>
<li>as input, you need a tibble (or a data.frame)</li>
<li>supply data, define aesthetic mapping, add layers, define axes, plot theme etc.</li>
<li>use the same theme (font size, line width etc.) across all figures in a project/paper/thesis</li>
<li>statistical tests</li>
<li>saving plots and source data</li>
</ul>

<img data-src="images/ggplot2-2177292224.png" class="r-stretch"></section>
<section id="aesthetics-plot-types-and-themes" class="slide level2">
<h2>Aesthetics, plot types and themes</h2>
<ul>
<li>need to define ‘aesthetics’, which variable goes to x, y axes, to colour, size, line thickness etc.</li>
<li>need to select plot type (geom_… boxplot, line, points etc.)</li>
<li>need to define ‘theme’ such as axis thickness, fonts, ticks, borders etc.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href=""></a>iris <span class="sc">%&gt;%</span>  </span>
<span id="cb56-2"><a href=""></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Sepal.Length, <span class="at">y =</span> Sepal.Width, <span class="at">color =</span> Species)) <span class="sc">+</span></span>
<span id="cb56-3"><a href=""></a>  <span class="fu">geom_boxplot</span>(<span class="at">notch =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb56-4"><a href=""></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-24-1.png" width="960" class="r-stretch"></section>
<section id="faceting" class="slide level2">
<h2>Faceting</h2>
<p><br></p>
<ul>
<li><p>Faceting means spreading the data across different panels</p></li>
<li><p>Need to define ‘facets’, a set of variables or expressions quoted by vars() defining faceting groups</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href=""></a>iris <span class="sc">%&gt;%</span>  </span>
<span id="cb57-2"><a href=""></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb57-3"><a href=""></a>    <span class="at">x =</span> Petal.Length, <span class="at">y =</span> Sepal.Width, </span>
<span id="cb57-4"><a href=""></a>    <span class="at">color =</span> Species, <span class="at">size =</span> Sepal.Width, <span class="at">shape =</span> Species) </span>
<span id="cb57-5"><a href=""></a>    ) <span class="sc">+</span></span>
<span id="cb57-6"><a href=""></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb57-7"><a href=""></a>  <span class="fu">facet_wrap</span>(<span class="fu">vars</span>(Species))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-25-1.png" width="960" class="r-stretch"></section>
<section id="resources-1" class="slide level2">
<h2>Resources</h2>
<p><br> https://ggplot2.tidyverse.org/</p>
<p>https://r4ds.hadley.nz/</p>
<ul>
<li>ggplot cheat sheet https://rstudio.github.io/cheatsheets/data-visualization.pdf</li>
</ul>

<img data-src="images/ggplot2-2177292224.png" class="r-stretch"></section>
<section id="plot-data---jose" class="slide level2">
<h2>Plot data - Jose</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href=""></a>plot_Jose1 <span class="ot">&lt;-</span> data_Jose <span class="sc">%&gt;%</span></span>
<span id="cb58-2"><a href=""></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> genotype, <span class="at">y =</span> length, <span class="at">fill =</span> <span class="fu">factor</span>(Treatment, <span class="at">level=</span><span class="fu">c</span>(<span class="st">'Control'</span>, <span class="st">'ABA'</span>, <span class="st">'Sulfate'</span>)), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb58-3"><a href=""></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb58-4"><a href=""></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb58-5"><a href=""></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#D55E00"</span>, <span class="st">"#E69F00"</span>, <span class="st">"#cccccc"</span>)) <span class="sc">+</span></span>
<span id="cb58-6"><a href=""></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> <span class="st">"Treatment"</span>)) </span>
<span id="cb58-7"><a href=""></a></span>
<span id="cb58-8"><a href=""></a>plot_Jose1  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-26-1.png" width="960"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href=""></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb59-2"><a href=""></a>plot_Jose2 <span class="ot">&lt;-</span> data_Jose <span class="sc">%&gt;%</span></span>
<span id="cb59-3"><a href=""></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> genotype, <span class="at">y =</span> length, <span class="at">fill =</span> <span class="fu">factor</span>(Treatment, <span class="at">level=</span><span class="fu">c</span>(<span class="st">'Control'</span>, <span class="st">'ABA'</span>, <span class="st">'Sulfate'</span>)), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb59-4"><a href=""></a>  <span class="fu">geom_violin</span>() <span class="sc">+</span></span>
<span id="cb59-5"><a href=""></a>  <span class="fu">geom_point</span>( <span class="at">position=</span><span class="fu">position_jitterdodge</span>(<span class="at">jitter.width =</span> <span class="fl">0.3</span>, <span class="at">dodge.width =</span> <span class="fl">0.9</span>), <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="fl">0.4</span>) <span class="sc">+</span></span>
<span id="cb59-6"><a href=""></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#D55E00"</span>, <span class="st">"#E69F00"</span>, <span class="st">"#aaaaaa"</span>, <span class="st">"#dddddd"</span>)) <span class="sc">+</span></span>
<span id="cb59-7"><a href=""></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> <span class="st">"Treatment"</span>)) </span>
<span id="cb59-8"><a href=""></a>plot_Jose2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-27-1.png" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="a-note-on-languages" class="slide level2">
<h2>A note on languages</h2>
<ul>
<li>you can also use python, bash etc in code blocks</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode numberSource python number-lines code-with-copy"><code class="sourceCode python"><span id="cb60-1"><a href=""></a><span class="im">import</span> matplotlib</span>
<span id="cb60-2"><a href=""></a><span class="bu">print</span>(matplotlib.__version__)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>3.3.4</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode numberSource python number-lines code-with-copy"><code class="sourceCode python"><span id="cb62-1"><a href=""></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb62-2"><a href=""></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb62-3"><a href=""></a></span>
<span id="cb62-4"><a href=""></a>r <span class="op">=</span> np.arange(<span class="dv">0</span>, <span class="dv">2</span>, <span class="fl">0.01</span>)</span>
<span id="cb62-5"><a href=""></a>theta <span class="op">=</span> <span class="dv">2</span> <span class="op">*</span> np.pi <span class="op">*</span> r</span>
<span id="cb62-6"><a href=""></a>fig, ax <span class="op">=</span> plt.subplots(</span>
<span id="cb62-7"><a href=""></a>  subplot_kw <span class="op">=</span> {<span class="st">'projection'</span>: <span class="st">'polar'</span>} </span>
<span id="cb62-8"><a href=""></a>)</span>
<span id="cb62-9"><a href=""></a>ax.plot(theta, r)</span>
<span id="cb62-10"><a href=""></a>ax.set_rticks([<span class="fl">0.5</span>, <span class="dv">1</span>, <span class="fl">1.5</span>, <span class="dv">2</span>])</span>
<span id="cb62-11"><a href=""></a>ax.grid(<span class="va">True</span>)</span>
<span id="cb62-12"><a href=""></a>plt.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/fig-polar-1.png" width="960" class="r-stretch quarto-figure-center"><p class="caption">
Figure&nbsp;1: A line plot on a polar axis
</p></section>
<section id="plot-data" class="slide level2">
<h2>Plot data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href=""></a>data_Ashwini_sel_M_SD <span class="sc">%&gt;%</span></span>
<span id="cb63-2"><a href=""></a>  <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span></span>
<span id="cb63-3"><a href=""></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> days, <span class="at">y =</span> dt_ct, <span class="at">fill =</span> gene )) <span class="sc">+</span></span>
<span id="cb63-4"><a href=""></a>  <span class="fu">geom_boxplot</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-30-3.png" width="960"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href=""></a>data_Ashwini_sel_M_SD <span class="sc">%&gt;%</span></span>
<span id="cb64-2"><a href=""></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> ct_value)) <span class="sc">+</span></span>
<span id="cb64-3"><a href=""></a>  <span class="fu">geom_histogram</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-30-4.png" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="plot-data-1" class="slide level2">
<h2>Plot data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href=""></a>plot_Ashwini_ct <span class="ot">&lt;-</span> data_Ashwini_sel_M_SD <span class="sc">%&gt;%</span></span>
<span id="cb65-2"><a href=""></a>  <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span></span>
<span id="cb65-3"><a href=""></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> gene, <span class="at">y =</span> ct_value, <span class="at">fill =</span> gene )) <span class="sc">+</span></span>
<span id="cb65-4"><a href=""></a>  <span class="fu">geom_boxplot</span>(<span class="at">na.rm =</span> <span class="cn">TRUE</span>) </span>
<span id="cb65-5"><a href=""></a>plot_Ashwini_ct</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-31-1.png" width="960" class="r-stretch"></section>
<section id="plot-the-synuclein-data" class="slide level2">
<h2>Plot the Synuclein data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href=""></a>plot_syn <span class="ot">&lt;-</span> tb_syn <span class="sc">%&gt;%</span></span>
<span id="cb66-2"><a href=""></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Time, <span class="at">y =</span> fluorescence, <span class="at">color =</span> condition)) <span class="sc">+</span></span>
<span id="cb66-3"><a href=""></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">'loess'</span>) <span class="sc">+</span></span>
<span id="cb66-4"><a href=""></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb66-5"><a href=""></a>plot_syn</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-32-1.png" width="960" class="r-stretch"></section>
<section id="read-and-preview-data-3" class="slide level2">
<h2>Read and preview data 3</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href=""></a>data_Anchel <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_excel</span>(<span class="st">"../analysis/data/240323 CIN Exp278 reporter assay - Anchel.xlsx"</span>)</span>
<span id="cb67-2"><a href=""></a><span class="fu">head</span>(data_Anchel)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 13
  Application: Tecan i-co…¹ ...2  ...3  ...4  Tecan i-control , 1.…² ...6   ...7
  &lt;chr&gt;                     &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;                  &lt;chr&gt; &lt;dbl&gt;
1 Device: infinite M1000    &lt;NA&gt;  &lt;NA&gt;  &lt;NA&gt;  Serial number: 812003… &lt;NA&gt;     NA
2 Firmware: V_2.09_04/2011… &lt;NA&gt;  &lt;NA&gt;  &lt;NA&gt;  MAI, V_2.09_04/2011_S… &lt;NA&gt;     NA
3 &lt;NA&gt;                      &lt;NA&gt;  &lt;NA&gt;  &lt;NA&gt;  &lt;NA&gt;                   &lt;NA&gt;     NA
4 Date:                     23.0… &lt;NA&gt;  &lt;NA&gt;  &lt;NA&gt;                   &lt;NA&gt;     NA
5 Time:                     11:4… &lt;NA&gt;  &lt;NA&gt;  &lt;NA&gt;                   &lt;NA&gt;     NA
6 &lt;NA&gt;                      &lt;NA&gt;  &lt;NA&gt;  &lt;NA&gt;  &lt;NA&gt;                   &lt;NA&gt;     NA
# ℹ abbreviated names: ¹​`Application: Tecan i-control`,
#   ²​`Tecan i-control , 1.10.4.0`
# ℹ 6 more variables: ...8 &lt;chr&gt;, ...9 &lt;chr&gt;, ...10 &lt;chr&gt;, ...11 &lt;dbl&gt;,
#   ...12 &lt;dbl&gt;, ...13 &lt;dbl&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href=""></a><span class="fu">glimpse</span>(data_Anchel)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 111
Columns: 13
$ `Application: Tecan i-control` &lt;chr&gt; "Device: infinite M1000", "Firmware: V_…
$ ...2                           &lt;chr&gt; NA, NA, NA, "23.03.2024", "11:48:52", N…
$ ...3                           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,…
$ ...4                           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,…
$ `Tecan i-control , 1.10.4.0`   &lt;chr&gt; "Serial number: 812003359", "MAI, V_2.0…
$ ...6                           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,…
$ ...7                           &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,…
$ ...8                           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,…
$ ...9                           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,…
$ ...10                          &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,…
$ ...11                          &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,…
$ ...12                          &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,…
$ ...13                          &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,…</code></pre>
</div>
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href=""></a><span class="fu">str</span>(data_Anchel)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>tibble [111 × 13] (S3: tbl_df/tbl/data.frame)
 $ Application: Tecan i-control: chr [1:111] "Device: infinite M1000" "Firmware: V_2.09_04/2011_S3LCE (May  2 2011/09.25.56)" NA "Date:" ...
 $ ...2                        : chr [1:111] NA NA NA "23.03.2024" ...
 $ ...3                        : chr [1:111] NA NA NA NA ...
 $ ...4                        : chr [1:111] NA NA NA NA ...
 $ Tecan i-control , 1.10.4.0  : chr [1:111] "Serial number: 812003359" "MAI, V_2.09_04/2011_S3LCE (May  2 2011/09.25.56)" NA NA ...
 $ ...6                        : chr [1:111] NA NA NA NA ...
 $ ...7                        : num [1:111] NA NA NA NA NA NA NA NA NA NA ...
 $ ...8                        : chr [1:111] NA NA NA NA ...
 $ ...9                        : chr [1:111] NA NA NA NA ...
 $ ...10                       : chr [1:111] NA NA NA NA ...
 $ ...11                       : num [1:111] NA NA NA NA NA NA NA NA NA NA ...
 $ ...12                       : num [1:111] NA NA NA NA NA NA NA NA NA NA ...
 $ ...13                       : num [1:111] NA NA NA NA NA NA NA NA NA NA ...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href=""></a><span class="fu">summary</span>(data_Anchel)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Application: Tecan i-control     ...2               ...3          
 Length:111                   Length:111         Length:111        
 Class :character             Class :character   Class :character  
 Mode  :character             Mode  :character   Mode  :character  
                                                                   
                                                                   
                                                                   
                                                                   
     ...4           Tecan i-control , 1.10.4.0     ...6          
 Length:111         Length:111                 Length:111        
 Class :character   Class :character           Class :character  
 Mode  :character   Mode  :character           Mode  :character  
                                                                 
                                                                 
                                                                 
                                                                 
      ...7               ...8               ...9              ...10          
 Min.   :   0.2220   Length:111         Length:111         Length:111        
 1st Qu.:   0.6881   Class :character   Class :character   Class :character  
 Median : 276.0000   Mode  :character   Mode  :character   Mode  :character  
 Mean   : 453.4537                                                           
 3rd Qu.: 979.5000                                                           
 Max.   :1213.0000                                                           
 NA's   :84                                                                  
     ...11               ...12               ...13         
 Min.   :    1.024   Min.   :    0.962   Min.   :   0.589  
 1st Qu.:    9.752   1st Qu.:   11.000   1st Qu.:   5.727  
 Median :  963.000   Median :  963.000   Median : 696.000  
 Mean   : 2821.028   Mean   : 2894.721   Mean   :1108.773  
 3rd Qu.: 1371.000   3rd Qu.: 1589.500   3rd Qu.:1063.000  
 Max.   :30845.000   Max.   :31016.000   Max.   :9274.000  
 NA's   :84          NA's   :84          NA's   :84        </code></pre>
</div>
</div>
</section>
<section>
<section id="add-significance-values-of-stats" class="title-slide slide level1 center">
<h1>Add significance values of stats</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href=""></a>plot_Ashwini_ct <span class="sc">+</span></span>
<span id="cb75-2"><a href=""></a>  <span class="fu">stat_compare_means</span>(</span>
<span id="cb75-3"><a href=""></a>    <span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(p.format)),</span>
<span id="cb75-4"><a href=""></a>    <span class="at">method =</span> <span class="st">"wilcox.test"</span>,</span>
<span id="cb75-5"><a href=""></a>    <span class="co">#ref.group = "Foxg1"</span></span>
<span id="cb75-6"><a href=""></a>    <span class="at">comparisons =</span> <span class="fu">list</span>( <span class="fu">c</span>(<span class="st">"GAPDH"</span>, <span class="st">"Nanog"</span>), <span class="fu">c</span>(<span class="st">"GAPDH"</span>, <span class="st">"oct4"</span>), <span class="fu">c</span>(<span class="st">"GAPDH"</span>, <span class="st">"pax6"</span>) )</span>
<span id="cb75-7"><a href=""></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-34-1.png" width="960" class="r-stretch"></section>
<section id="save-tidy-data-as-source-data-for-the-plotfigurepaper" class="slide level2">
<h2>Save tidy data as source data for the plot/figure/paper</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href=""></a><span class="fu">write_csv2</span>(data_Ashwini_sel_M_SD, <span class="st">"../manuscript/source_data/FigureX_Ashwini_source_data.csv"</span>)</span>
<span id="cb76-2"><a href=""></a><span class="co"># check</span></span>
<span id="cb76-3"><a href=""></a><span class="fu">read_csv2</span>(<span class="st">"../manuscript/source_data/data_Ashwini_sel_M_SD.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 105 × 1
   `gene,days,raw_data,ct_value,dt_ct,x2_dct,mean2dct,sd2dct`                   
   &lt;chr&gt;                                                                        
 1 Nanog,d0,J1,22.491,6.9559999999999995,0.008054439990584293,0.090867588244444…
 2 Nanog,d0,J2,19.877,4.395999999999999,0.0474976517625474,0.09086758824444471,…
 3 Nanog,d0,J3,19.164,2.418000000000003,0.18711537334908607,0.09086758824444471…
 4 Nanog,d7,J4,21.177,4.667999999999999,0.039336161672026825,0.0908675882444447…
 5 Nanog,d7,J5,21.812,4.934000000000001,0.03271282131170547,0.09086758824444471…
 6 Nanog,d7,J6,22.303,5.397000000000002,0.023732370153351993,0.0908675882444447…
 7 Nanog,d11,J7,22.93,7.436,0.005774853247163447,0.09086758824444471,0.14124081…
 8 Nanog,d11,J8,22.915,7.4129999999999985,0.00586765596461301,0.090867588244444…
 9 Nanog,d11,J9,17.831,1.2369999999999983,0.4242539508529486,0.0908675882444447…
10 Nanog,d15,J10,15.848,1.2520000000000007,0.41986574647207,0.09086758824444471…
# ℹ 95 more rows</code></pre>
</div>
</div>
</section>
<section id="format-plots-with-predefined-complete-ggplot2-themes" class="slide level2">
<h2>Format plots with predefined complete ggplot2 themes</h2>
<div class="columns">
<div class="column" style="width:50%;">
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href=""></a>plot_Jose1 <span class="sc">+</span></span>
<span id="cb78-2"><a href=""></a>  <span class="fu">theme_dark</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-36-1.png" width="960"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href=""></a>plot_Jose1 <span class="sc">+</span></span>
<span id="cb79-2"><a href=""></a>  <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-36-2.png" width="960"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href=""></a>plot_Jose1 <span class="sc">+</span></span>
<span id="cb80-2"><a href=""></a>  <span class="fu">theme_linedraw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-36-3.png" width="960"></p>
</figure>
</div>
</div>
</div>
</div><div class="column" style="width:50%;">
<div class="cell">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href=""></a>plot_Jose2 <span class="sc">+</span></span>
<span id="cb81-2"><a href=""></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-37-1.png" width="960"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href=""></a>plot_Jose2 <span class="sc">+</span></span>
<span id="cb82-2"><a href=""></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-37-2.png" width="960"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href=""></a>plot_Jose2 <span class="sc">+</span></span>
<span id="cb83-2"><a href=""></a>  <span class="fu">theme_light</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-37-3.png" width="960"></p>
</figure>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="format-plots-with-a-common-custom-theme" class="slide level2">
<h2>Format plots with a common custom theme()</h2>
<ul>
<li>Themes are a powerful way to customize the non-data components of your plots: i.e.&nbsp;titles, labels, fonts, background, gridlines, and legends.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href=""></a><span class="fu">args</span>(theme)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>function (..., line, rect, text, title, aspect.ratio, axis.title, 
    axis.title.x, axis.title.x.top, axis.title.x.bottom, axis.title.y, 
    axis.title.y.left, axis.title.y.right, axis.text, axis.text.x, 
    axis.text.x.top, axis.text.x.bottom, axis.text.y, axis.text.y.left, 
    axis.text.y.right, axis.text.theta, axis.text.r, axis.ticks, 
    axis.ticks.x, axis.ticks.x.top, axis.ticks.x.bottom, axis.ticks.y, 
    axis.ticks.y.left, axis.ticks.y.right, axis.ticks.theta, 
    axis.ticks.r, axis.minor.ticks.x.top, axis.minor.ticks.x.bottom, 
    axis.minor.ticks.y.left, axis.minor.ticks.y.right, axis.minor.ticks.theta, 
    axis.minor.ticks.r, axis.ticks.length, axis.ticks.length.x, 
    axis.ticks.length.x.top, axis.ticks.length.x.bottom, axis.ticks.length.y, 
    axis.ticks.length.y.left, axis.ticks.length.y.right, axis.ticks.length.theta, 
    axis.ticks.length.r, axis.minor.ticks.length, axis.minor.ticks.length.x, 
    axis.minor.ticks.length.x.top, axis.minor.ticks.length.x.bottom, 
    axis.minor.ticks.length.y, axis.minor.ticks.length.y.left, 
    axis.minor.ticks.length.y.right, axis.minor.ticks.length.theta, 
    axis.minor.ticks.length.r, axis.line, axis.line.x, axis.line.x.top, 
    axis.line.x.bottom, axis.line.y, axis.line.y.left, axis.line.y.right, 
    axis.line.theta, axis.line.r, legend.background, legend.margin, 
    legend.spacing, legend.spacing.x, legend.spacing.y, legend.key, 
    legend.key.size, legend.key.height, legend.key.width, legend.key.spacing, 
    legend.key.spacing.x, legend.key.spacing.y, legend.frame, 
    legend.ticks, legend.ticks.length, legend.axis.line, legend.text, 
    legend.text.position, legend.title, legend.title.position, 
    legend.position, legend.position.inside, legend.direction, 
    legend.byrow, legend.justification, legend.justification.top, 
    legend.justification.bottom, legend.justification.left, legend.justification.right, 
    legend.justification.inside, legend.location, legend.box, 
    legend.box.just, legend.box.margin, legend.box.background, 
    legend.box.spacing, panel.background, panel.border, panel.spacing, 
    panel.spacing.x, panel.spacing.y, panel.grid, panel.grid.major, 
    panel.grid.minor, panel.grid.major.x, panel.grid.major.y, 
    panel.grid.minor.x, panel.grid.minor.y, panel.ontop, plot.background, 
    plot.title, plot.title.position, plot.subtitle, plot.caption, 
    plot.caption.position, plot.tag, plot.tag.position, plot.tag.location, 
    plot.margin, strip.background, strip.background.x, strip.background.y, 
    strip.clip, strip.placement, strip.text, strip.text.x, strip.text.x.bottom, 
    strip.text.x.top, strip.text.y, strip.text.y.left, strip.text.y.right, 
    strip.switch.pad.grid, strip.switch.pad.wrap, complete = FALSE, 
    validate = TRUE) 
NULL</code></pre>
</div>
</div>
</section>
<section id="format-plots-with-a-common-custom-theme-1" class="slide level2">
<h2>Format plots with a common custom theme()</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href=""></a>theme_plots <span class="ot">&lt;-</span> <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb86-2"><a href=""></a>  <span class="fu">theme</span>(</span>
<span id="cb86-3"><a href=""></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb86-4"><a href=""></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb86-5"><a href=""></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb86-6"><a href=""></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb86-7"><a href=""></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb86-8"><a href=""></a>    <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="dv">7</span>, <span class="st">"mm"</span>)</span>
<span id="cb86-9"><a href=""></a>  )</span>
<span id="cb86-10"><a href=""></a></span>
<span id="cb86-11"><a href=""></a>plot_Ashwini_ct <span class="ot">&lt;-</span> plot_Ashwini_ct <span class="sc">+</span></span>
<span id="cb86-12"><a href=""></a>  theme_plots</span>
<span id="cb86-13"><a href=""></a>plot_Ashwini_ct</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-39-1.png" width="960"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href=""></a>plot_Jose1 <span class="ot">&lt;-</span> plot_Jose1 <span class="sc">+</span></span>
<span id="cb87-2"><a href=""></a>  theme_plots</span>
<span id="cb87-3"><a href=""></a>plot_Jose1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-39-2.png" width="960"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href=""></a>plot_Jose2 <span class="ot">&lt;-</span> plot_Jose2 <span class="sc">+</span></span>
<span id="cb88-2"><a href=""></a>  theme_plots</span>
<span id="cb88-3"><a href=""></a>plot_Jose2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-39-3.png" width="960"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href=""></a>plot_syn <span class="ot">&lt;-</span> plot_syn <span class="sc">+</span></span>
<span id="cb89-2"><a href=""></a>  theme_plots</span>
<span id="cb89-3"><a href=""></a>plot_syn</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-39-4.png" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="optional---save-plots" class="slide level2">
<h2>Optional - save plots</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href=""></a><span class="fu">ggsave</span>( <span class="st">"../analysis/pictures/plot_Jose1a.png"</span>,</span>
<span id="cb90-2"><a href=""></a>  <span class="at">limitsize =</span> <span class="cn">FALSE</span>,</span>
<span id="cb90-3"><a href=""></a>  <span class="at">units =</span> <span class="fu">c</span>(<span class="st">"px"</span>), plot_Jose1,</span>
<span id="cb90-4"><a href=""></a>  <span class="at">width =</span> <span class="dv">2400</span>, <span class="at">height =</span> <span class="dv">1400</span>, <span class="at">bg =</span> <span class="st">"white"</span></span>
<span id="cb90-5"><a href=""></a>)</span>
<span id="cb90-6"><a href=""></a></span>
<span id="cb90-7"><a href=""></a><span class="co"># save in a different size</span></span>
<span id="cb90-8"><a href=""></a><span class="fu">ggsave</span>( <span class="st">"../analysis/pictures/plot_Jose1b.png"</span>,</span>
<span id="cb90-9"><a href=""></a>  <span class="at">limitsize =</span> <span class="cn">FALSE</span>,</span>
<span id="cb90-10"><a href=""></a>  <span class="at">units =</span> <span class="fu">c</span>(<span class="st">"px"</span>), plot_Jose2,</span>
<span id="cb90-11"><a href=""></a>  <span class="at">width =</span> <span class="dv">2400</span>, <span class="at">height =</span> <span class="dv">2000</span>, <span class="at">bg =</span> <span class="st">"white"</span></span>
<span id="cb90-12"><a href=""></a>)</span>
<span id="cb90-13"><a href=""></a></span>
<span id="cb90-14"><a href=""></a></span>
<span id="cb90-15"><a href=""></a><span class="fu">ggsave</span>(</span>
<span id="cb90-16"><a href=""></a>  <span class="st">"../analysis/pictures/synuclein_plot.png"</span>, plot_syn, </span>
<span id="cb90-17"><a href=""></a>  <span class="at">bg =</span> <span class="st">"white"</span></span>
<span id="cb90-18"><a href=""></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="part-4---assembly-of-figures-with-cowplot-and-patchwork" class="slide level2">
<h2>Part 4 - Assembly of figures with cowplot and patchwork</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href=""></a>img1 <span class="ot">&lt;-</span> <span class="fu">readPNG</span>(<span class="st">"../analysis/pictures/plot_Jose1a.png"</span>)</span>
<span id="cb91-2"><a href=""></a>img2 <span class="ot">&lt;-</span> <span class="fu">readPNG</span>(<span class="st">"../analysis/pictures/plot_Jose1b.png"</span>)</span>
<span id="cb91-3"><a href=""></a>panel_JoseA <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> <span class="fu">draw_image</span>(img1)</span>
<span id="cb91-4"><a href=""></a>panel_JoseB <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> <span class="fu">draw_image</span>(img2)</span>
<span id="cb91-5"><a href=""></a></span>
<span id="cb91-6"><a href=""></a><span class="co">#define layout with textual representation</span></span>
<span id="cb91-7"><a href=""></a>layout <span class="ot">&lt;-</span> <span class="st">"</span></span>
<span id="cb91-8"><a href=""></a><span class="st">AB</span></span>
<span id="cb91-9"><a href=""></a><span class="st">CD"</span></span>
<span id="cb91-10"><a href=""></a></span>
<span id="cb91-11"><a href=""></a><span class="co">#assemble multipanel figure based on layout</span></span>
<span id="cb91-12"><a href=""></a>Figure_Jose <span class="ot">&lt;-</span> panel_JoseA <span class="sc">+</span> panel_JoseB <span class="sc">+</span> plot_Jose1 <span class="sc">+</span> plot_Jose2 <span class="sc">+</span></span>
<span id="cb91-13"><a href=""></a>  <span class="fu">plot_layout</span>(<span class="at">design =</span> layout, <span class="at">heights =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb91-14"><a href=""></a>  <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">'A'</span>) <span class="sc">&amp;</span> </span>
<span id="cb91-15"><a href=""></a>  <span class="fu">theme</span>(<span class="at">plot.tag =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face=</span><span class="st">'plain'</span>))</span>
<span id="cb91-16"><a href=""></a></span>
<span id="cb91-17"><a href=""></a><span class="co">#save figure as png and pdf</span></span>
<span id="cb91-18"><a href=""></a><span class="fu">ggsave</span>(</span>
<span id="cb91-19"><a href=""></a>  <span class="st">"../manuscript/figures/Figure_Jose.png"</span>, <span class="at">limitsize =</span> <span class="cn">FALSE</span>, </span>
<span id="cb91-20"><a href=""></a>  <span class="at">units =</span> <span class="fu">c</span>(<span class="st">"px"</span>), Figure_Jose, <span class="at">width =</span> <span class="dv">4000</span>, <span class="at">height =</span> <span class="dv">2000</span>, <span class="at">bg =</span> <span class="st">"white"</span></span>
<span id="cb91-21"><a href=""></a>  )</span>
<span id="cb91-22"><a href=""></a><span class="fu">ggsave</span>(</span>
<span id="cb91-23"><a href=""></a>  <span class="st">"../manuscript/figures/Figure_Jose.pdf"</span>, <span class="at">limitsize =</span> <span class="cn">FALSE</span>, </span>
<span id="cb91-24"><a href=""></a>  <span class="at">units =</span> <span class="fu">c</span>(<span class="st">"px"</span>), Figure_Jose, <span class="at">width =</span> <span class="dv">3000</span>, <span class="at">height =</span> <span class="dv">1600</span></span>
<span id="cb91-25"><a href=""></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>

<img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-42-1.png" width="2000" class="r-stretch"></section>
<section id="annotating-a-ggplot-object" class="slide level2">
<h2>Annotating a ggplot object</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href=""></a>plot_syn_ann <span class="ot">&lt;-</span> plot_syn <span class="sc">+</span></span>
<span id="cb92-2"><a href=""></a>  <span class="fu">annotate</span>(<span class="st">"segment"</span>, <span class="at">x =</span> <span class="dv">20</span>, <span class="at">xend =</span> <span class="dv">50</span>, <span class="at">y =</span> <span class="dv">1</span>, <span class="at">yend =</span> <span class="dv">1</span>, <span class="at">linewidth =</span> <span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb92-3"><a href=""></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">34</span>, <span class="at">y =</span> <span class="dv">300</span>, <span class="at">label =</span> <span class="st">"30 sec"</span>, <span class="at">size =</span> <span class="dv">3</span>)</span>
<span id="cb92-4"><a href=""></a>plot_syn_ann</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-43-1.png" width="960" class="r-stretch"></section>
<section id="annotating-an-image" class="slide level2">
<h2>Annotating an image</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href=""></a><span class="co">#read images</span></span>
<span id="cb93-2"><a href=""></a>img_INNOS <span class="ot">&lt;-</span> magick<span class="sc">::</span><span class="fu">image_read</span>(<span class="st">"../analysis/pictures/INNOS_synapses.png"</span>)</span>
<span id="cb93-3"><a href=""></a></span>
<span id="cb93-4"><a href=""></a><span class="co">#define arrow endpoints </span></span>
<span id="cb93-5"><a href=""></a>arrow <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">x1 =</span> <span class="fl">0.95</span>, <span class="at">x2 =</span> <span class="fl">0.95</span>, <span class="at">y1 =</span> <span class="fl">0.8</span>, <span class="at">y2 =</span> <span class="fl">0.9</span>)</span>
<span id="cb93-6"><a href=""></a></span>
<span id="cb93-7"><a href=""></a><span class="co">#add text labels</span></span>
<span id="cb93-8"><a href=""></a>panel_INNOS <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> </span>
<span id="cb93-9"><a href=""></a>  <span class="fu">draw_image</span>(img_INNOS) <span class="sc">+</span></span>
<span id="cb93-10"><a href=""></a>  <span class="fu">draw_label</span>(<span class="st">"INNOS"</span>, <span class="at">x =</span> <span class="fl">0.3</span>, <span class="at">y =</span> <span class="fl">0.99</span>, <span class="at">size =</span> <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb93-11"><a href=""></a>  <span class="fu">draw_label</span>(<span class="st">"NS plexus"</span>, <span class="at">x =</span> <span class="fl">0.485</span>, <span class="at">y =</span> <span class="fl">0.59</span>, <span class="at">size =</span> <span class="dv">8</span>) <span class="sc">+</span></span>
<span id="cb93-12"><a href=""></a>  <span class="fu">draw_label</span>(<span class="st">"outgoing"</span>, <span class="at">x =</span> <span class="fl">0.9</span>, <span class="at">y =</span> <span class="fl">0.45</span>, <span class="at">size =</span> <span class="dv">10</span>, <span class="at">color=</span><span class="st">'#E69F00'</span>) <span class="sc">+</span></span>
<span id="cb93-13"><a href=""></a>  <span class="fu">draw_label</span>(<span class="st">"incoming"</span>, <span class="at">x =</span> <span class="fl">0.89</span>, <span class="at">y =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">10</span>, <span class="at">color=</span><span class="st">'#0072B2'</span>) <span class="sc">+</span></span>
<span id="cb93-14"><a href=""></a>  <span class="fu">draw_label</span>(<span class="st">"D"</span>, <span class="at">x =</span> <span class="fl">0.95</span>, <span class="at">y =</span> <span class="fl">0.93</span>, <span class="at">size =</span> <span class="dv">6</span>) <span class="sc">+</span></span>
<span id="cb93-15"><a href=""></a>  <span class="fu">draw_label</span>(<span class="st">"V"</span>, <span class="at">x =</span> <span class="fl">0.95</span>, <span class="at">y =</span> <span class="fl">0.77</span>, <span class="at">size =</span> <span class="dv">6</span>) <span class="sc">+</span></span>
<span id="cb93-16"><a href=""></a>  <span class="fu">draw_label</span>(<span class="st">"*"</span>, <span class="at">x =</span> <span class="fl">0.5</span>, <span class="at">y =</span> <span class="fl">0.29</span>, <span class="at">color=</span><span class="st">'black'</span>,<span class="at">size =</span> <span class="dv">18</span>,<span class="at">fontface=</span><span class="st">'plain'</span>) <span class="sc">+</span></span>
<span id="cb93-17"><a href=""></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> x1, <span class="at">y =</span> y1, <span class="at">xend =</span> x2, <span class="at">yend =</span> y2), <span class="at">data =</span> arrow, </span>
<span id="cb93-18"><a href=""></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">ends =</span> <span class="st">"both"</span>, <span class="at">type =</span> <span class="st">"closed"</span>, <span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.1</span>,<span class="st">"cm"</span>)),</span>
<span id="cb93-19"><a href=""></a>               <span class="at">lineend =</span> <span class="st">"butt"</span>,</span>
<span id="cb93-20"><a href=""></a>               <span class="at">linejoin =</span> <span class="st">"mitre"</span>,</span>
<span id="cb93-21"><a href=""></a>               <span class="at">arrow.fill =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="fl">0.2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="annotating-an-image-1" class="slide level2">
<h2>Annotating an image</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href=""></a><span class="co">#define layout</span></span>
<span id="cb94-2"><a href=""></a>layout <span class="ot">&lt;-</span> <span class="st">"AB"</span></span>
<span id="cb94-3"><a href=""></a></span>
<span id="cb94-4"><a href=""></a><span class="co">#assemble multipanel figure based on layout</span></span>
<span id="cb94-5"><a href=""></a>Figure_INNOS <span class="ot">&lt;-</span> plot_syn_ann <span class="sc">+</span> panel_INNOS <span class="sc">+</span></span>
<span id="cb94-6"><a href=""></a>  <span class="fu">plot_layout</span>(<span class="at">design =</span> layout, <span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb94-7"><a href=""></a>  <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">'A'</span>) <span class="sc">&amp;</span> </span>
<span id="cb94-8"><a href=""></a>  <span class="fu">theme</span>(<span class="at">plot.tag =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face=</span><span class="st">'plain'</span>))</span>
<span id="cb94-9"><a href=""></a></span>
<span id="cb94-10"><a href=""></a><span class="co">#save figure as png</span></span>
<span id="cb94-11"><a href=""></a><span class="fu">ggsave</span>(</span>
<span id="cb94-12"><a href=""></a>  <span class="st">"../manuscript/figures/Figure_INNOS.png"</span>, <span class="at">limitsize =</span> <span class="cn">FALSE</span>,</span>
<span id="cb94-13"><a href=""></a>  <span class="at">units =</span> <span class="fu">c</span>(<span class="st">"px"</span>), Figure_INNOS, </span>
<span id="cb94-14"><a href=""></a>  <span class="at">width =</span> <span class="dv">3000</span>, <span class="at">height =</span> <span class="dv">1000</span>,</span>
<span id="cb94-15"><a href=""></a>  <span class="at">bg =</span> <span class="st">"white"</span></span>
<span id="cb94-16"><a href=""></a>  )</span>
<span id="cb94-17"><a href=""></a></span>
<span id="cb94-18"><a href=""></a><span class="co">#save figure as pdf</span></span>
<span id="cb94-19"><a href=""></a><span class="fu">ggsave</span>(</span>
<span id="cb94-20"><a href=""></a>  <span class="st">"../manuscript/figures/Figure_IHC.pdf"</span>, <span class="at">limitsize =</span> <span class="cn">FALSE</span>, </span>
<span id="cb94-21"><a href=""></a>  <span class="at">units =</span> <span class="fu">c</span>(<span class="st">"px"</span>), Figure_INNOS, <span class="at">width =</span> <span class="dv">3000</span>, <span class="at">height =</span> <span class="dv">1000</span></span>
<span id="cb94-22"><a href=""></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>

<img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-46-1.png" width="1500" class="r-stretch"></section>
<section id="adding-consistent-scale-bars" class="slide level2">
<h2>Adding consistent scale bars</h2>
<ul>
<li>you could add the scale bar directly on the image e.g., in ImageJ, however..</li>
<li>your scale bars may be misaligned and of varying thickness</li>
<li>you can use cowplot::draw_line instead</li>
<li>the x positions of the start and end of the line are defined as % of the panel width</li>
<li>if you know the width of the image (e.g., can add it to file name), it is easy to calculate the size of the scale bar</li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href=""></a><span class="co">#read images and make annotated panel</span></span>
<span id="cb95-2"><a href=""></a>panel_NOS2d_HCR <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> <span class="fu">draw_image</span>(<span class="fu">readPNG</span>(<span class="st">"../analysis/pictures/HCR-IHC_51_AP_NOS_actub_56um.png"</span>)) <span class="sc">+</span></span>
<span id="cb95-3"><a href=""></a>  <span class="fu">draw_label</span>(<span class="st">"in situ HCR"</span>, <span class="at">x =</span> <span class="fl">0.3</span>, <span class="at">y =</span> <span class="fl">0.99</span>, <span class="at">size =</span> <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb95-4"><a href=""></a>  <span class="fu">draw_label</span>(<span class="st">"NOS"</span>, <span class="at">x =</span> <span class="fl">0.12</span>, <span class="at">y =</span> <span class="fl">0.9</span>, <span class="at">color=</span><span class="st">"magenta"</span>, <span class="at">size =</span> <span class="dv">11</span>, <span class="at">fontface=</span><span class="st">"italic"</span>) <span class="sc">+</span></span>
<span id="cb95-5"><a href=""></a>  <span class="fu">draw_label</span>(<span class="st">"acTub"</span>, <span class="at">x =</span> <span class="fl">0.36</span>, <span class="at">y =</span> <span class="fl">0.9</span>, <span class="at">color=</span><span class="st">"green"</span>, <span class="at">size =</span> <span class="dv">11</span>, <span class="at">fontface=</span><span class="st">"plain"</span>) <span class="sc">+</span></span>
<span id="cb95-6"><a href=""></a>  <span class="fu">draw_line</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.46</span>), <span class="at">y =</span> <span class="fu">c</span>(<span class="fl">0.08</span>, <span class="fl">0.08</span>), <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb95-7"><a href=""></a>  <span class="fu">draw_label</span>(<span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">"20 "</span>, mu, <span class="st">" m"</span>)), <span class="at">x =</span> <span class="fl">0.28</span>, <span class="at">y =</span> <span class="fl">0.11</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">8</span>)</span>
<span id="cb95-8"><a href=""></a>  </span>
<span id="cb95-9"><a href=""></a></span>
<span id="cb95-10"><a href=""></a>panel_NIT_HCR <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> <span class="fu">draw_image</span>(<span class="fu">readPNG</span>(<span class="st">"../analysis/pictures/HCR_72_AP_NIT_94um.png"</span>)) <span class="sc">+</span></span>
<span id="cb95-11"><a href=""></a>  <span class="fu">draw_label</span>(<span class="st">"transgene + IHC"</span>, <span class="at">x =</span> <span class="fl">0.38</span>, <span class="at">y =</span> <span class="fl">0.99</span>, <span class="at">size =</span> <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb95-12"><a href=""></a>  <span class="fu">draw_label</span>(<span class="st">"NOSp::palmi-3xHA"</span>, <span class="at">x =</span> <span class="fl">0.34</span>, <span class="at">y =</span> <span class="fl">0.9</span>, <span class="at">color=</span><span class="st">"magenta"</span>, <span class="at">size =</span> <span class="dv">10</span>, <span class="at">fontface=</span><span class="st">"plain"</span>) <span class="sc">+</span></span>
<span id="cb95-13"><a href=""></a>  <span class="fu">draw_label</span>(<span class="st">"acTub"</span>, <span class="at">x =</span> <span class="fl">0.8</span>, <span class="at">y =</span> <span class="fl">0.9</span>, <span class="at">color=</span><span class="st">"green"</span>, <span class="at">size =</span> <span class="dv">10</span>, <span class="at">fontface=</span><span class="st">"plain"</span>) <span class="sc">+</span></span>
<span id="cb95-14"><a href=""></a>  <span class="fu">draw_line</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.31</span>), <span class="at">y =</span> <span class="fu">c</span>(<span class="fl">0.08</span>, <span class="fl">0.08</span>), <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="fl">0.5</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href=""></a><span class="co">#introduce gap in layout</span></span>
<span id="cb96-2"><a href=""></a>layout <span class="ot">&lt;-</span> <span class="st">"A#B"</span></span>
<span id="cb96-3"><a href=""></a></span>
<span id="cb96-4"><a href=""></a><span class="co">#assemble multipanel figure based on layout</span></span>
<span id="cb96-5"><a href=""></a>Figure_scalebars <span class="ot">&lt;-</span> panel_NOS2d_HCR <span class="sc">+</span> panel_NIT_HCR <span class="sc">+</span></span>
<span id="cb96-6"><a href=""></a>  <span class="fu">plot_layout</span>(<span class="at">design =</span> layout, <span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">0.03</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb96-7"><a href=""></a>  <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">'A'</span>) <span class="sc">&amp;</span> </span>
<span id="cb96-8"><a href=""></a>  <span class="fu">theme</span>(<span class="at">plot.tag =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face=</span><span class="st">'plain'</span>))</span>
<span id="cb96-9"><a href=""></a></span>
<span id="cb96-10"><a href=""></a><span class="co">#save figure as png</span></span>
<span id="cb96-11"><a href=""></a><span class="fu">ggsave</span>(</span>
<span id="cb96-12"><a href=""></a>  <span class="st">"../manuscript/figures/Figure_scalebars.png"</span>,</span>
<span id="cb96-13"><a href=""></a>  <span class="at">units =</span> <span class="fu">c</span>(<span class="st">"px"</span>), Figure_scalebars, </span>
<span id="cb96-14"><a href=""></a>  <span class="at">width =</span> <span class="dv">1700</span>, <span class="at">height =</span> <span class="dv">940</span>, <span class="at">bg =</span> <span class="st">"white"</span></span>
<span id="cb96-15"><a href=""></a>  )</span>
<span id="cb96-16"><a href=""></a></span>
<span id="cb96-17"><a href=""></a><span class="fu">image_read</span>(<span class="st">"../manuscript/figures/Figure_scalebars.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-48-1.png" width="850"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href=""></a><span class="fu">image_read</span>(<span class="st">"../manuscript/figures/Figure_scalebars.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-48-2.png" width="850"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="reading-tif-files-parsing-metadata-like-image-scale" class="slide level2">
<h2>Reading tif files, parsing metadata like image scale</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href=""></a>img_tif <span class="ot">&lt;-</span> magick<span class="sc">::</span><span class="fu">image_read</span>(<span class="st">"../analysis/pictures/INNOS_synapses.tif"</span>)</span>
<span id="cb98-2"><a href=""></a>img_scale <span class="ot">&lt;-</span> <span class="fu">image_info</span>(img_tif)<span class="sc">$</span>width<span class="sc">/</span><span class="fu">as.numeric</span>(<span class="fu">str_split</span>(<span class="fu">image_info</span>(img_tif)<span class="sc">$</span>density, <span class="st">"x"</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>])</span>
<span id="cb98-3"><a href=""></a>img_scale <span class="co">#not precise</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 29.62963</code></pre>
</div>
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href=""></a><span class="co"># alternatively use ijtiff</span></span>
<span id="cb100-2"><a href=""></a><span class="fu">library</span>(ijtiff)</span>
<span id="cb100-3"><a href=""></a>img_tif_ij <span class="ot">&lt;-</span> ijtiff<span class="sc">::</span><span class="fu">read_tags</span>(<span class="st">"../analysis/pictures/INNOS_synapses.tif"</span>)</span>
<span id="cb100-4"><a href=""></a>img_scale <span class="ot">&lt;-</span> img_tif_ij<span class="sc">$</span>frame1<span class="sc">$</span>width<span class="sc">/</span>img_tif_ij<span class="sc">$</span>frame1<span class="sc">$</span>x_resolution</span>
<span id="cb100-5"><a href=""></a>img_scale</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 30</code></pre>
</div>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href=""></a>panel_INNOS_tif <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> <span class="fu">draw_image</span>(img_tif)</span>
<span id="cb102-2"><a href=""></a></span>
<span id="cb102-3"><a href=""></a>Figure_INNOS_with_tif <span class="ot">&lt;-</span> plot_syn_ann <span class="sc">+</span> panel_INNOS_tif <span class="sc">+</span></span>
<span id="cb102-4"><a href=""></a>  <span class="fu">plot_layout</span>(<span class="at">design =</span> layout, <span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb102-5"><a href=""></a>  <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">'A'</span>) <span class="sc">&amp;</span> </span>
<span id="cb102-6"><a href=""></a>  <span class="fu">theme</span>(<span class="at">plot.tag =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face=</span><span class="st">'plain'</span>))</span>
<span id="cb102-7"><a href=""></a></span>
<span id="cb102-8"><a href=""></a><span class="co">#save figure as png</span></span>
<span id="cb102-9"><a href=""></a><span class="fu">ggsave</span>(</span>
<span id="cb102-10"><a href=""></a>  <span class="st">"../manuscript/figures/Figure_INNOS_with_tif.png"</span>, <span class="at">limitsize =</span> <span class="cn">FALSE</span>,</span>
<span id="cb102-11"><a href=""></a>  <span class="at">units =</span> <span class="fu">c</span>(<span class="st">"px"</span>), Figure_INNOS_with_tif, </span>
<span id="cb102-12"><a href=""></a>  <span class="at">width =</span> <span class="dv">3000</span>, <span class="at">height =</span> <span class="dv">1000</span>, <span class="at">bg =</span> <span class="st">"white"</span></span>
<span id="cb102-13"><a href=""></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>

<img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-50-1.png" width="1500" class="r-stretch"></section>
<section id="fine-tuning-figure-size-and-gaps" class="slide level2">
<h2>Fine-tuning figure size and gaps</h2>
<h3 id="exercises">Exercises</h3>
<ul>
<li>save the figure in different sizes</li>
<li>introduce gap with # into layout, also need to define width of gap as say 0.05</li>
<li>change position of scalebar and scalebar legend</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href=""></a><span class="co">#no gap in layout</span></span>
<span id="cb103-2"><a href=""></a>layout1 <span class="ot">&lt;-</span> <span class="st">"AB"</span></span>
<span id="cb103-3"><a href=""></a></span>
<span id="cb103-4"><a href=""></a><span class="co">#assemble multipanel figure based on layout</span></span>
<span id="cb103-5"><a href=""></a>Figure_scalebars <span class="ot">&lt;-</span> panel_NOS2d_HCR <span class="sc">+</span> panel_NIT_HCR <span class="sc">+</span></span>
<span id="cb103-6"><a href=""></a>  <span class="fu">plot_layout</span>(<span class="at">design =</span> layout1, <span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb103-7"><a href=""></a>  <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">'A'</span>) <span class="sc">&amp;</span> </span>
<span id="cb103-8"><a href=""></a>  <span class="fu">theme</span>(<span class="at">plot.tag =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face=</span><span class="st">'plain'</span>))</span>
<span id="cb103-9"><a href=""></a></span>
<span id="cb103-10"><a href=""></a><span class="co">#save figure as png</span></span>
<span id="cb103-11"><a href=""></a><span class="fu">ggsave</span>(</span>
<span id="cb103-12"><a href=""></a>  <span class="st">"../manuscript/figures/Figure_scalebars_no_gap.png"</span>, </span>
<span id="cb103-13"><a href=""></a>  <span class="at">limitsize =</span> <span class="cn">FALSE</span>,</span>
<span id="cb103-14"><a href=""></a>  <span class="at">units =</span> <span class="fu">c</span>(<span class="st">"px"</span>), Figure_scalebars, </span>
<span id="cb103-15"><a href=""></a>  <span class="at">width =</span> <span class="dv">1700</span>, <span class="at">height =</span> <span class="dv">940</span>,</span>
<span id="cb103-16"><a href=""></a>  <span class="at">bg =</span> <span class="st">"white"</span></span>
<span id="cb103-17"><a href=""></a>  )</span>
<span id="cb103-18"><a href=""></a></span>
<span id="cb103-19"><a href=""></a><span class="fu">image_read</span>(<span class="st">"../manuscript/figures/Figure_scalebars_no_gap.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-51-1.png" width="850"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href=""></a><span class="co">#introduce gap in layout</span></span>
<span id="cb104-2"><a href=""></a>layout2 <span class="ot">&lt;-</span> <span class="st">"A#B"</span></span>
<span id="cb104-3"><a href=""></a></span>
<span id="cb104-4"><a href=""></a><span class="co">#assemble multipanel figure based on layout</span></span>
<span id="cb104-5"><a href=""></a>Figure_scalebars <span class="ot">&lt;-</span> panel_NOS2d_HCR <span class="sc">+</span> panel_NIT_HCR <span class="sc">+</span></span>
<span id="cb104-6"><a href=""></a>  <span class="fu">plot_layout</span>(<span class="at">design =</span> layout2, <span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">0.03</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb104-7"><a href=""></a>  <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">'A'</span>) <span class="sc">&amp;</span> </span>
<span id="cb104-8"><a href=""></a>  <span class="fu">theme</span>(<span class="at">plot.tag =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face=</span><span class="st">'plain'</span>))</span>
<span id="cb104-9"><a href=""></a></span>
<span id="cb104-10"><a href=""></a><span class="co">#save figure as png</span></span>
<span id="cb104-11"><a href=""></a><span class="fu">ggsave</span>(</span>
<span id="cb104-12"><a href=""></a>  <span class="st">"../manuscript/figures/Figure_scalebars_gap.png"</span>, </span>
<span id="cb104-13"><a href=""></a>  <span class="at">limitsize =</span> <span class="cn">FALSE</span>,</span>
<span id="cb104-14"><a href=""></a>  <span class="at">units =</span> <span class="fu">c</span>(<span class="st">"px"</span>), Figure_scalebars, </span>
<span id="cb104-15"><a href=""></a>  <span class="at">width =</span> <span class="dv">1700</span>, <span class="at">height =</span> <span class="dv">940</span>,</span>
<span id="cb104-16"><a href=""></a>  <span class="at">bg =</span> <span class="st">"white"</span></span>
<span id="cb104-17"><a href=""></a>  )</span>
<span id="cb104-18"><a href=""></a></span>
<span id="cb104-19"><a href=""></a><span class="fu">image_read</span>(<span class="st">"../manuscript/figures/Figure_scalebars_gap.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-51-2.png" width="850"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="more-complex-figure-layouts" class="slide level2">
<h2>More complex figure layouts</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href=""></a><span class="co">#read images and make annotated panel</span></span>
<span id="cb105-2"><a href=""></a>panel_Platy <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> <span class="fu">draw_image</span>(<span class="fu">readPNG</span>(<span class="st">"../analysis/pictures/Platynereis_SEM_inverted_nolabel.png"</span>))</span>
<span id="cb105-3"><a href=""></a>panel_NOS <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> <span class="fu">draw_image</span>(<span class="fu">readPNG</span>(<span class="st">"../analysis/pictures/HCR-IHC_51_AP_NOS_actub_56um.png"</span>))</span>
<span id="cb105-4"><a href=""></a>panel_FVRI <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> <span class="fu">draw_image</span>(<span class="fu">readPNG</span>(<span class="st">"../analysis/pictures/FVRIa_rhoPhall_31h_200um.png"</span>))</span>
<span id="cb105-5"><a href=""></a>panel_Jose <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> <span class="fu">draw_image</span>(<span class="fu">readPNG</span>(<span class="st">"../analysis/pictures/plot_Jose1b.png"</span>))</span>
<span id="cb105-6"><a href=""></a>panel_INNOS <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> <span class="fu">draw_image</span>(<span class="fu">readPNG</span>(<span class="st">"../analysis/pictures/INNOS_synapses.png"</span>))</span>
<span id="cb105-7"><a href=""></a>panel_NIT <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> <span class="fu">draw_image</span>(<span class="fu">readPNG</span>(<span class="st">"../analysis/pictures/IHC_55_AP_NITGC2_actub_61um.png"</span>))</span>
<span id="cb105-8"><a href=""></a>panel_DAF <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> <span class="fu">draw_image</span>(<span class="fu">readPNG</span>(<span class="st">"../analysis/pictures/DAFFM.png"</span>))</span>
<span id="cb105-9"><a href=""></a>panel_model <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> <span class="fu">draw_image</span>(<span class="fu">readPNG</span>(<span class="st">"../analysis/pictures/Magnitude_model_cPRC.png"</span>))</span>
<span id="cb105-10"><a href=""></a></span>
<span id="cb105-11"><a href=""></a><span class="co">#introduce gap in layout</span></span>
<span id="cb105-12"><a href=""></a>layout <span class="ot">&lt;-</span> <span class="st">"</span></span>
<span id="cb105-13"><a href=""></a><span class="st">AAAABBBBCCCC</span></span>
<span id="cb105-14"><a href=""></a><span class="st">AAAABBBBDDDD</span></span>
<span id="cb105-15"><a href=""></a><span class="st">############</span></span>
<span id="cb105-16"><a href=""></a><span class="st">EEEFFFGGGHHH</span></span>
<span id="cb105-17"><a href=""></a><span class="st">EEEFFFGGGHHH</span></span>
<span id="cb105-18"><a href=""></a><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="more-complex-figure-layouts-1" class="slide level2">
<h2>More complex figure layouts</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href=""></a><span class="co">#assemble multipanel figure based on layout</span></span>
<span id="cb106-2"><a href=""></a>Figure_complex <span class="ot">&lt;-</span> panel_Platy <span class="sc">+</span> panel_FVRI <span class="sc">+</span>  panel_NOS <span class="sc">+</span> </span>
<span id="cb106-3"><a href=""></a>  panel_NIT <span class="sc">+</span></span>
<span id="cb106-4"><a href=""></a>  panel_INNOS <span class="sc">+</span> panel_Jose <span class="sc">+</span> panel_DAF <span class="sc">+</span></span>
<span id="cb106-5"><a href=""></a>  panel_model <span class="sc">+</span></span>
<span id="cb106-6"><a href=""></a>  <span class="fu">plot_layout</span>(<span class="at">design =</span> layout, <span class="at">heights =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">0.05</span>, <span class="dv">1</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb106-7"><a href=""></a>  <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">'A'</span>) <span class="sc">&amp;</span> </span>
<span id="cb106-8"><a href=""></a>  <span class="fu">theme</span>(<span class="at">plot.tag =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face=</span><span class="st">'plain'</span>))</span>
<span id="cb106-9"><a href=""></a></span>
<span id="cb106-10"><a href=""></a><span class="co">#save figure as png</span></span>
<span id="cb106-11"><a href=""></a><span class="fu">ggsave</span>(</span>
<span id="cb106-12"><a href=""></a>  <span class="st">"../manuscript/figures/Figure_complex.png"</span>,</span>
<span id="cb106-13"><a href=""></a>  <span class="at">units =</span> <span class="fu">c</span>(<span class="st">"px"</span>), Figure_complex, </span>
<span id="cb106-14"><a href=""></a>  <span class="at">width =</span> <span class="dv">2600</span>, <span class="at">height =</span> <span class="dv">1700</span>, <span class="at">bg =</span> <span class="st">"white"</span></span>
<span id="cb106-15"><a href=""></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>

<img data-src="RStudio_paper_writing_course_presentation_part1_files/figure-revealjs/unnamed-chunk-54-1.png" width="1300" class="r-stretch"></section>
<section id="vector-graphics-and-drawings" class="slide level2">
<h2>Vector graphics and drawings</h2>
<p><br></p>
<ul>
<li>https://scidraw.io/ for high-quality CC-BY drawings</li>
<li>Inkscape (free) or Illustrator (license) for vector graphics</li>
<li>export as image, import into R workflow</li>
</ul>
</section>
<section id="image-saved-with-a-defined-resolution-dpi" class="slide level2">
<h2>Image saved with a defined resolution (dpi)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href=""></a><span class="fu">ggsave</span>(</span>
<span id="cb107-2"><a href=""></a>  <span class="st">"../manuscript/figures/Figure_complex_300dpi.png"</span>,</span>
<span id="cb107-3"><a href=""></a>  <span class="at">units =</span> <span class="fu">c</span>(<span class="st">"cm"</span>), Figure_complex, </span>
<span id="cb107-4"><a href=""></a>  <span class="at">width =</span> <span class="dv">22</span>, <span class="at">height =</span> <span class="dv">13</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">bg =</span> <span class="st">"white"</span></span>
<span id="cb107-5"><a href=""></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="part-5---writing-papers-and-reports-in-quartormarkdown" class="slide level2">
<h2>Part 5 - Writing papers and reports in Quarto/Rmarkdown</h2>
<ul>
<li><p>Powerful and versatile text editing with simple Markdown syntax or visual editor (similar to Word)</p></li>
<li><p>Easy insertion and formatting of references from the web or a .bib file</p></li>
<li><p>Supports equations, tables, code blocks</p></li>
<li><p>Directly insert your figures generated by your scripts in the same project</p></li>
<li><p>Can ‘knit’ into html, pdf or docx formats</p></li>
<li><p>-&gt; Open manuscript/Manuscript.qmd</p></li>
</ul>
</section>
<section id="quarto---documentation" class="slide level2">
<h2>Quarto - Documentation</h2>
<p><br></p>
<ul>
<li>Quarto enables you to weave together content and executable code into a finished presentation.</li>
<li>To learn more about Quarto presentations see <a href="https://quarto.org/docs/presentations/" class="uri">https://quarto.org/docs/presentations/</a>.</li>
</ul>
</section>
<section id="the-yaml-header" class="slide level2">
<h2>The YAML header</h2>
<ul>
<li>A YAML header contains YAML arguments, such as “title”, “author”, and “output”, demarcated by three dashes (-) on either end.</li>
<li>When you create a new Quarto doc or presentation, it is created with a short YAML</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href=""></a><span class="sc">---</span></span>
<span id="cb108-2"><a href=""></a>title<span class="sc">:</span> <span class="st">"Untitled"</span></span>
<span id="cb108-3"><a href=""></a>format<span class="sc">:</span> html</span>
<span id="cb108-4"><a href=""></a><span class="sc">---</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>There is a more complex YAML in manuscript/Manuscript.qmd</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href=""></a><span class="sc">---</span></span>
<span id="cb109-2"><a href=""></a>title<span class="sc">:</span> <span class="st">"Paper or report template"</span></span>
<span id="cb109-3"><a href=""></a>subtitle<span class="sc">:</span> <span class="er">|</span></span>
<span id="cb109-4"><a href=""></a>  \raggedright  Author <span class="dv">1</span><span class="sc">^</span><span class="dv">1</span>,<span class="dv">2</span>,\<span class="sc">*</span><span class="er">^</span> , Author <span class="dv">2</span><span class="sc">^</span><span class="dv">1</span>,<span class="dv">2</span><span class="sc">^</span>, Author <span class="dv">3</span><span class="sc">^</span><span class="dv">3</span><span class="sc">^</span>, Author <span class="dv">4</span>E<span class="sc">^</span><span class="dv">4</span><span class="sc">^</span> \newline<span class="sc">&lt;</span>br<span class="sc">&gt;</span><span class="er">^</span><span class="dv">1</span><span class="sc">^</span>Heidelberg University, Centre <span class="cf">for</span> Organismal <span class="fu">Studies</span> (COS), <span class="dv">69120</span> Heidelberg, Germany \newline<span class="sc">&lt;</span>br<span class="sc">&gt;</span><span class="er">^</span><span class="dv">2</span><span class="sc">^</span>Affiliation <span class="dv">2</span> \newline<span class="sc">&lt;</span>br<span class="sc">&gt;</span><span class="er">^</span><span class="dv">3</span><span class="sc">^</span>Affiliation <span class="dv">3</span> \newline<span class="sc">&lt;</span>br<span class="sc">&gt;</span><span class="er">^</span><span class="dv">4</span><span class="sc">^</span>Affiliation <span class="dv">4</span>\newline<span class="sc">&lt;</span>br<span class="sc">&gt;</span>  <span class="er">^</span>\<span class="sc">*</span><span class="er">^</span>Correspondence<span class="sc">:</span> xy<span class="sc">@</span>cos.uni<span class="sc">-</span>heidelberg.de</span>
<span id="cb109-5"><a href=""></a>mainfont<span class="sc">:</span> Arial</span>
<span id="cb109-6"><a href=""></a>monofont<span class="sc">:</span> DejaVu Sans Mono <span class="co">#Times New Roman</span></span>
<span id="cb109-7"><a href=""></a>fontsize<span class="sc">:</span> <span class="dv">11</span>pt</span>
<span id="cb109-8"><a href=""></a>linestretch<span class="sc">:</span> <span class="fl">1.2</span></span>
<span id="cb109-9"><a href=""></a>linkcolor<span class="sc">:</span> blue</span>
<span id="cb109-10"><a href=""></a>documentclass<span class="sc">:</span> article</span>
<span id="cb109-11"><a href=""></a>crossref<span class="sc">:</span></span>
<span id="cb109-12"><a href=""></a>  fig<span class="sc">-</span>title<span class="sc">:</span> Figure     <span class="co"># (default is "Figure")</span></span>
<span id="cb109-13"><a href=""></a>max<span class="sc">-</span>width<span class="sc">:</span> <span class="dv">100</span>%</span>
<span id="cb109-14"><a href=""></a>format<span class="sc">:</span></span>
<span id="cb109-15"><a href=""></a>  html<span class="sc">:</span> </span>
<span id="cb109-16"><a href=""></a>    embed<span class="sc">-</span>resources<span class="sc">:</span> true</span>
<span id="cb109-17"><a href=""></a>  pdf<span class="sc">:</span></span>
<span id="cb109-18"><a href=""></a>    pdf<span class="sc">-</span>engine<span class="sc">:</span> xelatex</span>
<span id="cb109-19"><a href=""></a>    number<span class="sc">-</span>sections<span class="sc">:</span> false</span>
<span id="cb109-20"><a href=""></a>    margin<span class="sc">-</span>left<span class="sc">:</span> <span class="fl">0.8</span>in</span>
<span id="cb109-21"><a href=""></a>    margin<span class="sc">-</span>right<span class="sc">:</span> <span class="fl">0.8</span>in</span>
<span id="cb109-22"><a href=""></a>    margin<span class="sc">-</span>top<span class="sc">:</span> <span class="fl">0.8</span>in</span>
<span id="cb109-23"><a href=""></a>    margin<span class="sc">-</span>bottom<span class="sc">:</span> <span class="fl">0.8</span>in</span>
<span id="cb109-24"><a href=""></a>    toc<span class="sc">:</span> true</span>
<span id="cb109-25"><a href=""></a>    keep<span class="sc">-</span>tex<span class="sc">:</span> true</span>
<span id="cb109-26"><a href=""></a>    include<span class="sc">-</span><span class="cf">in</span><span class="sc">-</span>header<span class="sc">:</span></span>
<span id="cb109-27"><a href=""></a>      text<span class="sc">:</span> <span class="er">|</span></span>
<span id="cb109-28"><a href=""></a>        \usepackage{lineno}</span>
<span id="cb109-29"><a href=""></a>    include<span class="sc">-</span>before<span class="sc">-</span>body<span class="sc">:</span></span>
<span id="cb109-30"><a href=""></a>      text<span class="sc">:</span> <span class="er">|</span></span>
<span id="cb109-31"><a href=""></a>        \linenumbers</span>
<span id="cb109-32"><a href=""></a>bibliography<span class="sc">:</span> references.bib <span class="co">#add bibliography in this file</span></span>
<span id="cb109-33"><a href=""></a>csl<span class="sc">:</span> elife.csl <span class="co">#change according to journal</span></span>
<span id="cb109-34"><a href=""></a><span class="sc">---</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="simple-syntax" class="slide level2">
<h2>Simple syntax</h2>
<ul>
<li><u>This is</u></li>
<li>A <strong>Quarto</strong> <em>document.</em></li>
<li>Quarto uses a simple <sup>Markdown</sup> formatting syntax for authoring HTML, PDF, and MS Word or .odt documents. - Here is some <strong>text in bold</strong> <em>italic</em> <u>underline</u> and <sup>superscript</sup>.</li>
</ul>
</section>
<section id="inserting-references" class="slide level2">
<h2>Inserting references</h2>
<p><br></p>
<ul>
<li>In Visual mode, go to ‘Insert’ -&gt; ‘Citation’</li>
<li>search for reference on pubmed etc.</li>
<li>‘Insert’ will add the ref to themanuscript/ bibliograpy.bib file (needs to be defined in the Yaml header)</li>
<li>in our doc, references will be formatted in the eLife style (defined in the Yaml header under: csl: elife.csl)</li>
<li>The elife.csl file is saved in the manuscript folder.</li>
<li>For a different format, download the respective .csl file (e.g., from the Zotero style repository <a href="https://www.zotero.org/styles" class="uri">https://www.zotero.org/styles</a>)</li>
<li>in ‘Source’ mode, you can directly insert the reference with <span class="citation" data-cites="ref_id">[@ref_id]</span> (preferred, faster method)</li>
</ul>
</section>
<section id="inserting-references-1" class="slide level2">
<h2>Inserting references</h2>
<p><br></p>
<ul>
<li>you can also add new refs to your .bib file with the command-line command:</li>
</ul>
<p>:$ curl -LH “Accept: application/x-bibtex” https://doi.org/10.7554/eLife.34550 &gt;&gt; ./references.bib <br></p>
<div class="callout callout-warning callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Warning</strong></p>
</div>
<div class="callout-content">
<ul>
<li>need to add references.bib to the YAML header as bibliography: references.bib</li>
<li>also define bibliography style in the YAML with csl: elife.csl #add .csl according to journal</li>
<li>if you use the command line, use &gt;&gt; and not &gt; otherwise you will overwrite your file!</li>
</ul>
</div>
</div>
</div>
</section>
<section id="inserting-figures" class="slide level2">
<h2>Inserting figures</h2>
<p><br></p>
<ul>
<li><p>we can now insert the figures we generated previously and saved under /manuscript/figures or /manuscript/figure_supplements</p></li>
<li><p>Figure title and legend can also be edited, as well as width</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href=""></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"../manuscript/figures/Figure1.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="../manuscript/figures/Figure1.png" style="width:100.0%" class="r-stretch quarto-figure-center"><p class="caption"><strong>Figure 1. A figure</strong> (A) A nice picture. (B) legend. (C) <br> (D)</p></section>
<section id="adding-equations" class="slide level2">
<h2>Adding equations</h2>
<p><br></p>
<ul>
<li>Equations can also be inserted, Insert -&gt; Display Math:</li>
</ul>
<p><span class="math display">\[
\bar{X} = \frac{\sum_{i=1}^{n} x_{i}}{n}
\]</span> ## Sourcing code and working with variable</p>
<ul>
<li>We can use the ‘r variable’ syntax to add variables calculated in the sourced code directly to the text</li>
<li>This way if the underlying dataset changes, we just reprocess a calculation and insert the result directly into our manuscript</li>
<li>the script is sourced and it runs but the output is not included in the knitted output. But we can access the variables defined in the sourced script simply by adding ` r var_name ` between ` backticks, in this case max_PRC value is 21 (now this number comes from our sourced script).</li>
</ul>
</section>
<section id="tables" class="slide level2">
<h2>Tables</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href=""></a><span class="co">#render built-in iris table with tinytable</span></span>
<span id="cb111-2"><a href=""></a>tinytable<span class="sc">::</span><span class="fu">tt</span>(<span class="fu">head</span>(iris))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<!-- preamble start -->

    <script>
      function styleCell_zmaqdpl1fvmxhu1crqae(i, j, css_id) {
        var table = document.getElementById("tinytable_zmaqdpl1fvmxhu1crqae");
        table.rows[i].cells[j].classList.add(css_id);
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_zmaqdpl1fvmxhu1crqae');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_zmaqdpl1fvmxhu1crqae(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_zmaqdpl1fvmxhu1crqae");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
window.addEventListener('load', function () { styleCell_zmaqdpl1fvmxhu1crqae(0, 0, 'tinytable_css_id4yy9oo000hv38u7lxyhk') })
window.addEventListener('load', function () { styleCell_zmaqdpl1fvmxhu1crqae(0, 1, 'tinytable_css_id4yy9oo000hv38u7lxyhk') })
window.addEventListener('load', function () { styleCell_zmaqdpl1fvmxhu1crqae(0, 2, 'tinytable_css_id4yy9oo000hv38u7lxyhk') })
window.addEventListener('load', function () { styleCell_zmaqdpl1fvmxhu1crqae(0, 3, 'tinytable_css_id4yy9oo000hv38u7lxyhk') })
window.addEventListener('load', function () { styleCell_zmaqdpl1fvmxhu1crqae(0, 4, 'tinytable_css_id4yy9oo000hv38u7lxyhk') })
    </script>

    <style>
    .table td.tinytable_css_id4yy9oo000hv38u7lxyhk, .table th.tinytable_css_id4yy9oo000hv38u7lxyhk {  border-bottom: solid 0.1em #d3d8dc; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_zmaqdpl1fvmxhu1crqae" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Sepal.Length</th>
                <th scope="col">Sepal.Width</th>
                <th scope="col">Petal.Length</th>
                <th scope="col">Petal.Width</th>
                <th scope="col">Species</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>5.1</td>
                  <td>3.5</td>
                  <td>1.4</td>
                  <td>0.2</td>
                  <td>setosa</td>
                </tr>
                <tr>
                  <td>4.9</td>
                  <td>3.0</td>
                  <td>1.4</td>
                  <td>0.2</td>
                  <td>setosa</td>
                </tr>
                <tr>
                  <td>4.7</td>
                  <td>3.2</td>
                  <td>1.3</td>
                  <td>0.2</td>
                  <td>setosa</td>
                </tr>
                <tr>
                  <td>4.6</td>
                  <td>3.1</td>
                  <td>1.5</td>
                  <td>0.2</td>
                  <td>setosa</td>
                </tr>
                <tr>
                  <td>5.0</td>
                  <td>3.6</td>
                  <td>1.4</td>
                  <td>0.2</td>
                  <td>setosa</td>
                </tr>
                <tr>
                  <td>5.4</td>
                  <td>3.9</td>
                  <td>1.7</td>
                  <td>0.4</td>
                  <td>setosa</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href=""></a><span class="co">#create some data</span></span>
<span id="cb112-2"><a href=""></a>mydata <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb112-3"><a href=""></a>  <span class="at">gene =</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">"B"</span>),</span>
<span id="cb112-4"><a href=""></a>  <span class="at">expression =</span> <span class="fu">c</span>(<span class="fl">1.1</span>, <span class="fl">0.4</span>)</span>
<span id="cb112-5"><a href=""></a>  )</span>
<span id="cb112-6"><a href=""></a></span>
<span id="cb112-7"><a href=""></a><span class="co">#render table</span></span>
<span id="cb112-8"><a href=""></a><span class="fu">tt</span>(mydata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<!-- preamble start -->

    <script>
      function styleCell_4amdc0oibyvj8pyp2r4l(i, j, css_id) {
        var table = document.getElementById("tinytable_4amdc0oibyvj8pyp2r4l");
        table.rows[i].cells[j].classList.add(css_id);
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_4amdc0oibyvj8pyp2r4l');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_4amdc0oibyvj8pyp2r4l(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_4amdc0oibyvj8pyp2r4l");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
window.addEventListener('load', function () { styleCell_4amdc0oibyvj8pyp2r4l(0, 0, 'tinytable_css_idedqei3iwpnv72cerggsg') })
window.addEventListener('load', function () { styleCell_4amdc0oibyvj8pyp2r4l(0, 1, 'tinytable_css_idedqei3iwpnv72cerggsg') })
    </script>

    <style>
    .table td.tinytable_css_idedqei3iwpnv72cerggsg, .table th.tinytable_css_idedqei3iwpnv72cerggsg {  border-bottom: solid 0.1em #d3d8dc; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_4amdc0oibyvj8pyp2r4l" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">gene</th>
                <th scope="col">expression</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>A</td>
                  <td>1.1</td>
                </tr>
                <tr>
                  <td>B</td>
                  <td>0.4</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href=""></a><span class="co">#read data</span></span>
<span id="cb113-2"><a href=""></a>resources_table <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xls</span>(<span class="st">"../analysis/data/Key_resource_table.xls"</span>)</span>
<span id="cb113-3"><a href=""></a></span>
<span id="cb113-4"><a href=""></a><span class="fu">tt</span>(</span>
<span id="cb113-5"><a href=""></a>  resources_table, <span class="at">caption =</span> <span class="st">"Key Resources Table"</span>, </span>
<span id="cb113-6"><a href=""></a>  <span class="at">theme =</span> <span class="st">"grid"</span>, <span class="at">width =</span> <span class="dv">1</span></span>
<span id="cb113-7"><a href=""></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<!-- preamble start -->

    <script>
      function styleCell_204f6taev9gnlbrsnqck(i, j, css_id) {
        var table = document.getElementById("tinytable_204f6taev9gnlbrsnqck");
        table.rows[i].cells[j].classList.add(css_id);
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_204f6taev9gnlbrsnqck');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_204f6taev9gnlbrsnqck(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_204f6taev9gnlbrsnqck");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
    </script>

    <style>
    </style>
    <div class="container">
      <table class="table table-bordered" id="tinytable_204f6taev9gnlbrsnqck" style="table-layout: fixed; width: 100% !important; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        </thead><caption>Key Resources Table</caption>
              <tbody><tr>
                <th scope="col">Reagent type (species) or resource</th>
                <th scope="col">Designation</th>
                <th scope="col">Source or reference</th>
                <th scope="col">Identifiers</th>
                <th scope="col">Additional information</th>
              </tr>
        
        
        </tbody><tbody>
                <tr>
                  <td>biological sample (N. vectensis)    </td>
                  <td>larval, juvenile and adult N. vectensis       </td>
                  <td>Specimens obtained form the Marine Invertebrate Culture Unit of the University of Exeter</td>
                  <td>N/A     </td>
                  <td>NA                                                                                                              </td>
                </tr>
                <tr>
                  <td>biological sample (cDNA)            </td>
                  <td>cDNA obtained from N. vectensis               </td>
                  <td>this study                                                                              </td>
                  <td>N/A     </td>
                  <td>RNA extracted with Trizol and cDNA synthesized with cDNA synthesis kit according to manufacturers recommendation</td>
                </tr>
                <tr>
                  <td>biological sample (peptide extract) </td>
                  <td>peptide extracts obtained from N. vectensis   </td>
                  <td>this study                                                                              </td>
                  <td>N/A     </td>
                  <td>Peptides extracted from N. vectensis according to protocol explained in Material and Methods                    </td>
                </tr>
                <tr>
                  <td>genetic reagent (cDNA synthesis)    </td>
                  <td>SuperScript™ III First-Strand Synthesis System</td>
                  <td>Invitrogen (from ThermoFisher)                                                          </td>
                  <td>18080051</td>
                  <td>NA                                                                                                              </td>
                </tr>
                <tr>
                  <td>genetic reagent (Polymerase)        </td>
                  <td>Q5® Hot Start High-Fidelity DNA Polymerase&nbsp;   </td>
                  <td>New England Biolabs                                                                     </td>
                  <td>M0493L  </td>
                  <td>NA                                                                                                              </td>
                </tr>
                <tr>
                  <td>genetic reagent (DNA assembly)      </td>
                  <td>NEBuilder®&nbsp;HiFi DNA Assembly Master Mix       </td>
                  <td>New England Biolabs                                                                     </td>
                  <td>E2621L  </td>
                  <td>NA                                                                                                              </td>
                </tr>
                <tr>
                  <td>genetic reagent (restriction enzyme)</td>
                  <td>EcoRV restriction enzyme                      </td>
                  <td>New England Biolabs                                                                     </td>
                  <td>R3195L  </td>
                  <td>NA                                                                                                              </td>
                </tr>
                <tr>
                  <td>genetic reagent (restriction enzyme)</td>
                  <td>Afl2 restriction enzyme                       </td>
                  <td>New England Biolabs                                                                     </td>
                  <td>R0520L  </td>
                  <td>NA                                                                                                              </td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</section>
<section id="part-6---collaborative-code-and-paper-writing-with-rstudio-and-github" class="slide level2">
<h2>Part 6 - Collaborative code and paper writing with Rstudio and GitHub</h2>
<p><br></p>
<ul>
<li>GitHub www.github.com is a platform for software development and version control</li>
<li>owned by Microsoft since 2018</li>
<li>developers can have public and private repositories</li>
<li>can share repositories privately with collaborators</li>
<li>integrates well with Rstudio</li>
<li>uses git protocol that is open source and also used by other code-hosting sites</li>
</ul>
</section>
<section id="the-workflow" class="slide level2">
<h2>The workflow</h2>
<ul>
<li>Create a new project or clone a template project on GitHub</li>
<li>In Rstudio, New Project -&gt; Version control -&gt; add GitHub repo URL and save locally</li>
<li>Change the code</li>
<li>You can ‘commit’ and then ‘push’ changes to GitHub</li>
<li>Your collaborators can be added on GitHub</li>
<li>Collaborators can ‘pull’ to their own computer and ‘push’ their own changes</li>
<li>If two developers push changes to the same repo, this can lead to conflict</li>
<li>This can be managed by starting new ‘branches’</li>
<li>Branches can be ‘merged’ to the original ‘master’ branch</li>
<li>A collaborator can issue a ‘pull request’ to merge their branch to the master</li>
</ul>
</section>
<section id="collaborations---sharing-and-commenting" class="slide level2">
<h2>Collaborations - sharing and commenting</h2>
<p><br></p>
<ul>
<li>Adding comments directly to the Quarto file with &lt;!–# my comment –&gt; (these will not be knitted to the output)</li>
<li>share as .docx and ask collaborators to use track changes (then why did we take this course…)</li>
<li>collaborate on the same GitHub repository, use branches and pull requests (only makes sense if the collaborators are very involved)</li>
<li>in principle you could use an Rproject for data, code and figures, and write text+refs on Gdocs+Paperpile or Overleaf+bib</li>
<li>share commentable html through your github or other website</li>
<li>we will use ‘https://hypothes.is/’ to add commenting functionality to an html</li>
<li>you need an account (easy and free)</li>
<li>we use a https://github.com/hypothesis/client script that is sourced at the end of the quarto file by &lt;script src=“hypothes.is/embed.js” async&gt; to make this work</li>
</ul>
<div class="callout callout-warning callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Warning</strong></p>
</div>
<div class="callout-content">
<ul>
<li>you need to host the html on a website for sharing</li>
</ul>
</div>
</div>
</div>
</section>
<section id="publication---github-and-zenodo-integration" class="slide level2">
<h2>Publication - GitHub and Zenodo integration</h2>
<p><br></p>
<ul>
<li>GitHub is not a publishing platform</li>
<li>The owner could delete or make private the entire repo one day after publishing</li>
<li>To stably publish your code and source data repository with a DOI ( globally unique and persistent identifier), you can use Zenodo</li>
<li>Launched in 2013 by CERN</li>
<li>Upload and link your GitHub repo to Zenodo</li>
</ul>
<p><img data-src="images/zenodo-gradient-1000.png" class="absolute" style="top: 260px; left: 550px; height: 100px; "></p>
</section>
<section id="citation" class="slide level2">
<h2>Citation</h2>
<p><br></p>
<ul>
<li>add/update a citation (CITATION.cff) file to your repo</li>
<li>include the Zenodo DOI</li>
<li>also cite your Zenodo repository in your manuscript</li>
<li>see <a href="https://zenodo.org/records/10978753">example</a></li>
<li>Make releases of your GitHub repo (e.g.&nbsp;initial submission, version of record), see <a href="https://github.com/JekelyLab/Thiel_Yanez_Nematostella/tree/v3.0.0">example</a></li>
<li>The releases will be shown on Zenodo and are citeable</li>
</ul>
</section>
<section id="add-readme-file-to-your-repository" class="slide level2">
<h2>Add README file to your repository</h2>
<p><br></p>
<ul>
<li>open the README file (README.Rmd)</li>
<li>README.md is generated from README.Rmd. Please edit that file</li>
<li>modify the file according to your project</li>
<li>add citation of the Zenodo repo with DOI</li>
<li>add License, Acknowledgements etc.</li>
</ul>
<div class="quarto-auto-generated-content">
<div class="footer footer-default">

</div>
</div>
</section></section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/plugin/reveal-chalkboard/plugin.js"></script>
  <script src="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="RStudio_paper_writing_course_presentation_part1_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': true,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleChalkboard(event)\"><kbd>b</kbd> Toggle Chalkboard</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleNotesCanvas(event)\"><kbd>c</kbd> Toggle Notes Canvas</a></li>\n<li class=\"slide-tool-item\" data-item=\"6\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.downloadDrawings(event)\"><kbd>d</kbd> Download Drawings</a></li>\n<li class=\"slide-tool-item\" data-item=\"7\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'chalkboard': {"buttons":false},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: false,

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealChalkboard, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        text: function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
        const config = {
          allowHTML: true,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start',
        };
        if (contentFn) {
          config.content = contentFn;
        }
        if (onTriggerFn) {
          config.onTrigger = onTriggerFn;
        }
        if (onUntriggerFn) {
          config.onUntrigger = onUntriggerFn;
        }
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>